<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Analysis and Visualization module | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="qc-module.html"/>
<link rel="next" href="learn-more.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#from-bam-files"><i class="fa fa-check"></i><b>4.1</b> From BAM files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1.1</b> runVarbin()</a></li>
<li class="chapter" data-level="4.1.2" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin-modules"><i class="fa fa-check"></i><b>4.1.2</b> runVarbin() modules</a></li>
<li class="chapter" data-level="4.1.3" data-path="pre-processing.html"><a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds"><i class="fa fa-check"></i><b>4.1.3</b> From external count data or user-defined scaffolds</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findaneuploidcells"><i class="fa fa-check"></i><b>5.2</b> findAneuploidCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#findoutliers"><i class="fa fa-check"></i><b>5.3</b> findOutliers()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>6</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>6.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>6.2</b> plotRatio()</a></li>
<li class="chapter" data-level="6.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>6.3</b> runUmap()</a></li>
<li class="chapter" data-level="6.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>6.4</b> plotUmap()</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>6.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>6.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="6.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>6.5.2</b> findClusters()</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo"><i class="fa fa-check"></i><b>6.6</b> runPhylo()</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotphylo"><i class="fa fa-check"></i><b>6.7</b> plotPhylo()</a></li>
<li class="chapter" data-level="6.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcinteger"><i class="fa fa-check"></i><b>6.8</b> calcInteger()</a></li>
<li class="chapter" data-level="6.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>6.9</b> plotHeatmap()</a></li>
<li class="chapter" data-level="6.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>6.10</b> plotFreq()</a></li>
<li class="chapter" data-level="6.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>6.11</b> calcConsensus()</a></li>
<li class="chapter" data-level="6.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>6.12</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="6.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>6.13</b> plotGeneCopy()</a></li>
<li class="chapter" data-level="6.14" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotalluvial"><i class="fa fa-check"></i><b>6.14</b> plotAlluvial()</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learn-more.html"><a href="learn-more.html"><i class="fa fa-check"></i><b>7</b> Learn more</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learn-more.html"><a href="learn-more.html#copy-number-data-from-short-reads-sequencing"><i class="fa fa-check"></i><b>7.1</b> Copy Number data from short reads sequencing</a></li>
<li class="chapter" data-level="7.2" data-path="learn-more.html"><a href="learn-more.html#principle-lm"><i class="fa fa-check"></i><b>7.2</b> Principle</a></li>
<li class="chapter" data-level="7.3" data-path="learn-more.html"><a href="learn-more.html#varbin-lm"><i class="fa fa-check"></i><b>7.3</b> VarBin</a></li>
<li class="chapter" data-level="7.4" data-path="learn-more.html"><a href="learn-more.html#gccor-lm"><i class="fa fa-check"></i><b>7.4</b> GC correction</a></li>
<li class="chapter" data-level="7.5" data-path="learn-more.html"><a href="learn-more.html#mergelevels-lm"><i class="fa fa-check"></i><b>7.5</b> Merge Levels</a></li>
<li class="chapter" data-level="7.6" data-path="learn-more.html"><a href="learn-more.html#infercn-lm"><i class="fa fa-check"></i><b>7.6</b> How do we infer copy numbers</a></li>
<li class="chapter" data-level="7.7" data-path="learn-more.html"><a href="learn-more.html#segmentation-lm"><i class="fa fa-check"></i><b>7.7</b> Segmentation</a>
<ul>
<li class="chapter" data-level="7.7.1" data-path="learn-more.html"><a href="learn-more.html#cbs-lm"><i class="fa fa-check"></i><b>7.7.1</b> CBS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>8</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis-and-visualization-module" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Analysis and Visualization module</h1>
<p>The analysis and visualization module from CopyKit work in synergy to help you analyze and assess the results.</p>
<div id="plotmetrics" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> plotMetrics()</h2>
<p><code>plotMetrics()</code> can plot any information available within <code>colData()</code>.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="analysis-and-visualization-module.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tumor))</span></code></pre></div>
<pre><code>##  [1] &quot;sample&quot;                &quot;reads_assigned_bins&quot;   &quot;reads_unmapped&quot;        &quot;reads_duplicates&quot;     
##  [5] &quot;reads_multimapped&quot;     &quot;reads_unassigned&quot;      &quot;reads_ambiguous&quot;       &quot;reads_total&quot;          
##  [9] &quot;percentage_duplicates&quot; &quot;overdispersion&quot;        &quot;breakpoint_count&quot;      &quot;is_aneuploid&quot;         
## [13] &quot;find_normal_cv&quot;        &quot;cell_corr_value&quot;       &quot;outlier&quot;</code></pre>
<p><code>colData</code> information is provided to the argument <code>metric</code>.</p>
<p>The argument label, if supplied, will color the points based on that information.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="analysis-and-visualization-module.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMetrics</span>(tumor, <span class="at">metric =</span> <span class="fu">c</span>(<span class="st">&quot;overdispersion&quot;</span>, </span>
<span id="cb91-2"><a href="analysis-and-visualization-module.html#cb91-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;breakpoint_count&quot;</span>,</span>
<span id="cb91-3"><a href="analysis-and-visualization-module.html#cb91-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_total&quot;</span>,</span>
<span id="cb91-4"><a href="analysis-and-visualization-module.html#cb91-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_duplicates&quot;</span>,</span>
<span id="cb91-5"><a href="analysis-and-visualization-module.html#cb91-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_assigned_bins&quot;</span>,</span>
<span id="cb91-6"><a href="analysis-and-visualization-module.html#cb91-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;percentage_duplicates&quot;</span>),</span>
<span id="cb91-7"><a href="analysis-and-visualization-module.html#cb91-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="st">&quot;percentage_duplicates&quot;</span>)</span></code></pre></div>
<pre><code>## Coloring by: percentage_duplicates</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="plotratio" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> plotRatio()</h2>
<p>It is important to visualize that the segmentation behaved as expected and closely follows the ratios.<br />
This helps to verify that the segmentation is correct and visually assess the data quality.</p>
<p>Ratio plots are a great tool for this. <code>plotRatio()</code> has two different modes.</p>
<p>When the input is the CopyKit object, an interactive app will open, allowing selection of which cell will be visualized.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="analysis-and-visualization-module.html#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_ratio_shiny.png" title="fig:" alt="Ratio Plot App" />
</center>
<p>If a sample name is provided to <code>plotRatio()</code>, only the plot for the selected cell will be shown.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="analysis-and-visualization-module.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_sample-1.png" width="672" /></p>
</div>
<div id="runumap" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> runUmap()</h2>
<p><code>runUmap()</code> generates a <a href="https://umap-learn.readthedocs.io/en/latest/">UMAP</a> embedding.
The resulting embedding is stored in the <code>reducedDim</code> slot.</p>
<p><code>runUmap()</code> is an important pre-processing step to the <code>findClusters()</code> feature.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="analysis-and-visualization-module.html#cb95-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: SingleCellExperiment::reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<p>Additional arguments to control umap parameters can be passed on to<code>runUmap()</code> with the ‘…’ argument.</p>
<p>The full list of additional arguments that can be passed on to <code>uwot::umap</code> with the ‘…’ argument can be seen in the <a href="https://cran.r-project.org/web/packages/uwot/uwot.pdf">uwot manual</a> and a guide to their influence on clustering can be found in the <a href="https://umap-learn.readthedocs.io/en/latest/clustering.html">umap webpage</a></p>
</div>
<div id="plotumap" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> plotUmap()</h2>
<p><code>plotUmap()</code> can be used to visualize the reduced dimensional embedding.
<code>plotUmap</code> can be colored by any element of the colData with the argument ‘label.’</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="analysis-and-visualization-module.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>An example on how to color the points by elements of colData is later shown in this guide.</p>
</div>
<div id="clustering" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Clustering</h2>
<p><code>findClusters()</code> uses the reduced dimensional embedding resulting from <code>runUmap()</code> to perform clustering at two levels (superclones and subclones).</p>
<p>When clustering for superclones <code>findClusters()</code> creates a graph representation of the reduced dimension embedding using a shared nearest neighbor algorithm (SNN).
From this graph the connected components are extracted and, generally, represent high-level structures that share large copy number events.</p>
<p>At a more fine-grained resolution, CopyKit can also be used to detect subclones. To do so, the umap embedding is used as the pre-processing step.</p>
<p>Different clustering algorithms can be used to detect subclones:</p>
<ol style="list-style-type: decimal">
<li><p><a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">hdbscan</a> (default)</p></li>
<li><p><a href="https://www.nature.com/articles/s41598-019-41695-z">Leiden</a></p></li>
<li><p><a href="https://arxiv.org/abs/0803.0476">Louvain</a></p></li>
</ol>
<p>The hdbscan method is recommended and has beenp reviously successfully applied in the work from Laks <em>et al.</em> and Minussi <em>et al.</em>.</p>
<div id="findsugestedk" class="section level3" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> findSugestedK()</h3>
<p>To help with parametrization, CopyKit provides the helper function <code>findSuggestedK()</code>. The function
<code>findSuggestedK</code> bootstraps clustering over a range of k values, and returns the value that maximizes the jaccard similarity. By default, the metric being maximized is <em>median</em> (argument <code>metric</code>).
While <code>findSuggestedK</code> does not guarantee optimal clustering, it does provides a guide that maximizes cluster stability.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="analysis-and-visualization-module.html#cb102-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(tumor)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 10 with median jaccard similarity of: 1</code></pre>
<p>We can inspect the results of <code>findSuggestedK()</code> with <code>plotSuggestedK()</code>.</p>
<p>The default geom plots a boxplot of each k assessed during the grid search. The red point inside each boxplot represents the mean jaccard similarity for a given k.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="analysis-and-visualization-module.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_boxplot-1.png" width="672" /></p>
<p>If the argument <code>geom</code> is set to <em>tile</em>, <code>plotSuggestedK()</code> plots a heatmap where each row is a detected subclone, each column is a k assessed during the grid search and the color represents the jaccard similarity for a given clone.</p>
<p>Clones not detected in a given k are presented in gray.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="analysis-and-visualization-module.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;tile&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_tile-1.png" width="672" /></p>
<p>It is also possible to visualize the relation between subclone sizes with the jaccard similarity setting geom to <em>scatterplot</em></p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="analysis-and-visualization-module.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;scatterplot&#39;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_scatter-1.png" width="576" /></p>
<p>The suggested value is stored into the metadata and can be accessed at:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="analysis-and-visualization-module.html#cb110-1" aria-hidden="true" tabindex="-1"></a>S4Vectors<span class="sc">::</span><span class="fu">metadata</span>(tumor)<span class="sc">$</span>suggestedK</span></code></pre></div>
<pre><code>## [1] 10</code></pre>
</div>
<div id="findclusters" class="section level3" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> findClusters()</h3>
<p>To run <code>findClusters()</code> we can provide the CopyKit object to the function.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="analysis-and-visualization-module.html#cb112-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 10</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 4 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<p>If <code>findSuggestedK()</code> was run prior, and the argument ‘k_subclones’ is not provided, <code>findClusters()</code> will automatically use the value resulting from <code>findSuggestedK()</code> that was stored into the metadata:</p>
<p>k values can be manually controlled with ‘k_superclones’ and ‘k_subclones.’
<em>NOTE:</em> k_superclones and k_subclones can be used concurrently. Only k_subclones is a mandatory field.</p>
<p>For example (not run):</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="analysis-and-visualization-module.html#cb117-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor,</span>
<span id="cb117-2"><a href="analysis-and-visualization-module.html#cb117-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_superclones =</span> <span class="dv">30</span>,</span>
<span id="cb117-3"><a href="analysis-and-visualization-module.html#cb117-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_subclones =</span> <span class="dv">15</span>)</span></code></pre></div>
<p>If the method used for clustering was <em>hdbscan</em>, it is possible that a subgroup of <a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">outliers</a> is identified. Outliers are added to subgroup <code>c0</code> and may be removed by subsetting the CopyKit object.
Copykit will report if any cell was classified as c0 in an informative message after running <code>findClusters()</code></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="analysis-and-visualization-module.html#cb118-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> tumor[,<span class="fu">colData</span>(tumor)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span></code></pre></div>
<p>We can use different plotting functions to visualize clustering results. For example <code>plotUmap()</code> can be used with the argument ‘label’:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="analysis-and-visualization-module.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_umap_subclones-1.png" width="672" /></p>
</div>
</div>
<div id="runphylo" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> runPhylo()</h2>
<p>To run a phylogenetic analysis of cells’ copy number profiles, use the function <code>runPhylo()</code>.
Available methods are <a href="https://pubmed.ncbi.nlm.nih.gov/3447015/">Neighbor Joining</a> and <a href="https://pubmed.ncbi.nlm.nih.gov/8412650/">Balanced Minimum evolution</a>.</p>
<p>The resulting tree is stored within the CopyKit object in the phylo slot:</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="analysis-and-visualization-module.html#cb122-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runPhylo</span>(tumor, <span class="at">metric =</span> <span class="st">&#39;manhattan&#39;</span>)</span></code></pre></div>
<pre><code>## Using ratio data...</code></pre>
<pre><code>## Calculating distance matrix</code></pre>
<pre><code>## Creating neighbor-joining tree.</code></pre>
<pre><code>## Access slot with copykit::phylo(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
</div>
<div id="plotphylo" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> plotPhylo()</h2>
<p>To visualize the resulting phylogenetic trees from <code>runPhylo()</code> we can use the <code>plotPhylo()</code> function.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="analysis-and-visualization-module.html#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo-1.png" width="672" /></p>
<p><code>plotPhylo()</code> can use any element of the colData to color the leaves of the tree.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="analysis-and-visualization-module.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo_label-1.png" width="672" /></p>
</div>
<div id="calcinteger" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> calcInteger()</h2>
<p>The true underlying copy number states for a given region of the genome are given in integer states.
We can use the segment ratio means to make an inference of the copy number state for a given genomic region.
Keep in mind that those are inferences and, as such, are subjected to errors of different sources, like erroneous inference or rounding errors.</p>
<p>Segment ratio means can be scaled to integer values with <code>calcInteger()</code>.</p>
<p>CopyKit supports different methods of calculating integer copy number profiles.</p>
<p>To calculate computational ploidies, CopyKit adopts the <a href="">scquantum</a> package. To use scquantum just set the argument method to <em>scquantum</em>.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="analysis-and-visualization-module.html#cb130-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">method =</span> <span class="st">&#39;scquantum&#39;</span>)</span></code></pre></div>
<p>By setting the argument <code>method</code> to <em>fixed</em>, a fixed value of ploidy (generally determined using Flow Cytometry) will scale all cells.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="analysis-and-visualization-module.html#cb131-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">ploidy_value =</span> <span class="fl">4.3</span>)</span></code></pre></div>
<pre><code>## Scaling ratio values by ploidy value 4.3</code></pre>
<p>Alternatively, if different ploidies are needed for different cells, a vector containing the ploidies can be passed on to <code>calcInteger()</code>, as long as it has an equal length and sample order as <code>colData()</code>.</p>
<p>The integer values are stored in the slot <code>integer</code> that can be accessed with the function <code>assay()</code>.</p>
<p>If the assay <em>integer</em> exists, <code>plotRatio()</code> will use it to plot the integer copy number values as a secondary axis.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="analysis-and-visualization-module.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_integer-1.png" width="672" /></p>
</div>
<div id="plotheatmap" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> plotHeatmap()</h2>
<p>To visualize copy number profiles with a heatmap we can use <code>plotHeatmap()</code>.</p>
<p>The heatmap can be annotated with elements of colData.</p>
<p>To order subclones, one option is to calculate a consensus phylogeny, explained in later sections:</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="analysis-and-visualization-module.html#cb134-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor)</span>
<span id="cb134-2"><a href="analysis-and-visualization-module.html#cb134-2" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runConsensusPhylo</span>(tumor)</span></code></pre></div>
<p>To plot the heatmap.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="analysis-and-visualization-module.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_heatmap_subclones-1.png" width="672" /></p>
<p>Genes can be annotated to a heatmap in <code>plotHeatmap()</code> with the <em>genes</em> argument.
The genes argument is a vector of gene HUGO symbols:</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="analysis-and-visualization-module.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;BRAF&quot;</span>, <span class="st">&quot;MYC&quot;</span>))</span></code></pre></div>
<p>To plot integer copy number heatmaps pass the argument <code>assay = 'integer'</code>, importantly the integer matrix must be in the <code>'integer'</code>slot.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="analysis-and-visualization-module.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ht_int-1.png" width="672" /></p>
<p>New information can be added to <code>colData()</code> and used in conjunction with the plotting functions.</p>
<p>The example dataset has macro-spatial information.
The information is encoded in the sample name by the letter L followed by a number.</p>
<p>We can extract that information and add it as an extra column to the metadata:</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="analysis-and-visualization-module.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tumor)<span class="sc">$</span>spatial_info <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(<span class="fu">colData</span>(tumor)<span class="sc">$</span>sample, <span class="st">&quot;L[0-9]&quot;</span>)</span></code></pre></div>
<p>Once the information has been added, we can use it to color the umap by their spatial information:</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="analysis-and-visualization-module.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: spatial_info.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/umap_spatial-1.png" width="672" /></p>
<p>It is also possible to annotate the heatmap with that information:
The ‘label’ argument for <code>plotHeatmap()</code> can add as many annotations as specified by the user as long as they are elements in <code>colData()</code> of the CopyKit object.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="analysis-and-visualization-module.html#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;spatial_info&quot;</span>, <span class="st">&quot;subclones&quot;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/heat_spatial-1.png" width="672" /></p>
</div>
<div id="plotfreq" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> plotFreq()</h2>
<p>Computing the frequencies of genomic gain or losses across the genome can be useful to visualize differences between groups. This can be done with <code>plotFreq()</code>.
For every region of the genome, <code>plotFreq()</code> will calculate the frequency of gain or losses according to a threshold across all samples.
The thresholds are controlled with the arguments <code>low_threshold</code> (values below will be counted as genomic losses) and <code>high_threshold</code> (values above will be counted as genomic gains).</p>
<p>Ideally thresholds will be set according to the ploidy of the sample.
For example, the sample presented throughout this documentation has a ploidy of approximately 4.3, therefore a gain or loss of one copy number for this sample should have a difference in the segment ratio value of 1/4.3 (~0.23).
This way, we will set the thresholds accordingly to a difference of plus or minus 0.2 (leaving some room for error), compared to a segment ratio of 1.</p>
<p>The argument <code>assay</code> can be provided to pass on the <code>integer</code> assay instead of the <code>segment_ratios</code> assay (adjust thresholds accordingly).</p>
<p>Two geoms are available <code>area</code> or <code>line</code>.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="analysis-and-visualization-module.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotFreq</span>(tumor, <span class="at">low_threshold =</span> <span class="fl">0.8</span>, <span class="at">high_threshold =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_freq_sample-1.png" width="672" /></p>
<p>Group information can be set to any categorical element of the <code>colData()</code> and is provided with the argument <code>label</code>.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="analysis-and-visualization-module.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotFreq</span>(tumor, </span>
<span id="cb147-2"><a href="analysis-and-visualization-module.html#cb147-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">low_threshold =</span> <span class="fl">0.8</span>, </span>
<span id="cb147-3"><a href="analysis-and-visualization-module.html#cb147-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">high_threshold =</span> <span class="fl">1.2</span>, </span>
<span id="cb147-4"><a href="analysis-and-visualization-module.html#cb147-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_freq_label-1.png" width="672" /></p>
</div>
<div id="calcconsensus" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> calcConsensus()</h2>
<p>Consensus sequences can help visualize the different segments across subclones. To calculate consensus matrices we can use <code>calcConsensus()</code>.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="analysis-and-visualization-module.html#cb148-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor)</span></code></pre></div>
<p><code>plotHeatmap()</code> can plot a consensus heatmap:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="analysis-and-visualization-module.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_heat_consensus-1.png" width="672" /></p>
<p><code>plotHeatmap()</code> can annotate the consensus heatmap with information from the metadata as long as label is the same as the information used to build the consensus matrix:</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="analysis-and-visualization-module.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">group =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_heat_freq_bar-1.png" width="672" /></p>
<p>By default <code>calcConsensus()</code> uses the subclones information to calculate a consensus for each subclone.
Any element of the <code>colData()</code> can be used to calculate the consensus.</p>
<p><strong>Note:</strong> Consensus matrices can be calculated from the integer assay. Importantly, the integer matrix must be in the <code>assay(tumor, 'integer')</code> slot. Check <code>calcInteger()</code> for more info.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="analysis-and-visualization-module.html#cb153-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor, <span class="at">consensus_by =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
</div>
<div id="plotconsensusline" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> plotConsensusLine()</h2>
<p>To compare the differences among subclones, <code>plotConsensusLine()</code> opens an interactive app where the consensus sequences are plotted as lines.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="analysis-and-visualization-module.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotConsensusLine</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_consensus_shiny.png" title="fig:" alt="Plot Consensus Line" />
</center>
</div>
<div id="plotgenecopy" class="section level2" number="6.13">
<h2><span class="header-section-number">6.13</span> plotGeneCopy()</h2>
<p>To check copy number states across of genes we can use <code>plotGeneCopy()</code>. Two different geoms: “swarm” (default) or “violin” can be applied.</p>
<p>As with other plotting functions, points can be colored with the argument ‘label.’</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="analysis-and-visualization-module.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb155-2"><a href="analysis-and-visualization-module.html#cb155-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb155-3"><a href="analysis-and-visualization-module.html#cb155-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb155-4"><a href="analysis-and-visualization-module.html#cb155-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb155-5"><a href="analysis-and-visualization-module.html#cb155-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb155-6"><a href="analysis-and-visualization-module.html#cb155-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb155-7"><a href="analysis-and-visualization-module.html#cb155-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb155-8"><a href="analysis-and-visualization-module.html#cb155-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb155-9"><a href="analysis-and-visualization-module.html#cb155-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb155-10"><a href="analysis-and-visualization-module.html#cb155-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb155-11"><a href="analysis-and-visualization-module.html#cb155-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb155-12"><a href="analysis-and-visualization-module.html#cb155-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Coloring by: spatial_info</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>A barplot geom is also provided to visualize the integer data as a frequency barplot for each gene:</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="analysis-and-visualization-module.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb157-2"><a href="analysis-and-visualization-module.html#cb157-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb157-3"><a href="analysis-and-visualization-module.html#cb157-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb157-4"><a href="analysis-and-visualization-module.html#cb157-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb157-5"><a href="analysis-and-visualization-module.html#cb157-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb157-6"><a href="analysis-and-visualization-module.html#cb157-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb157-7"><a href="analysis-and-visualization-module.html#cb157-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb157-8"><a href="analysis-and-visualization-module.html#cb157-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb157-9"><a href="analysis-and-visualization-module.html#cb157-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb157-10"><a href="analysis-and-visualization-module.html#cb157-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb157-11"><a href="analysis-and-visualization-module.html#cb157-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb157-12"><a href="analysis-and-visualization-module.html#cb157-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom =</span> <span class="st">&#39;barplot&#39;</span>,</span>
<span id="cb157-13"><a href="analysis-and-visualization-module.html#cb157-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_gene_copy_integer-1.png" width="672" /></p>
</div>
<div id="plotalluvial" class="section level2" number="6.14">
<h2><span class="header-section-number">6.14</span> plotAlluvial()</h2>
<p>To visualize frequencies across elements of the metadata we can use <code>plotAlluvial()</code></p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="analysis-and-visualization-module.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAlluvial</span>(tumor, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;subclones&quot;</span>, <span class="st">&quot;spatial_info&quot;</span>))</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="qc-module.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="learn-more.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
