<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Analysis and Visualization module | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="qc-module.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1</b> runVarbin()</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findnormalcells"><i class="fa fa-check"></i><b>5.2</b> findNormalCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#filtercells"><i class="fa fa-check"></i><b>5.3</b> filterCells()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>6</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>6.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>6.2</b> plotRatio()</a></li>
<li class="chapter" data-level="6.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>6.3</b> runUmap()</a></li>
<li class="chapter" data-level="6.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>6.4</b> plotUmap()</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>6.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>6.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="6.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>6.5.2</b> findClusters()</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo"><i class="fa fa-check"></i><b>6.6</b> runPhylo()</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotphylo"><i class="fa fa-check"></i><b>6.7</b> plotPhylo()</a></li>
<li class="chapter" data-level="6.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcinteger"><i class="fa fa-check"></i><b>6.8</b> calcInteger()</a></li>
<li class="chapter" data-level="6.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>6.9</b> plotHeatmap()</a></li>
<li class="chapter" data-level="6.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>6.10</b> calcConsensus()</a></li>
<li class="chapter" data-level="6.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>6.11</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="6.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>6.12</b> plotGeneCopy()</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>7</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis-and-visualization-module" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Analysis and Visualization module</h1>
<p>The analysis and visualization module from CopyKit work in synergy to help you analyze and assess the results from your day.</p>
<p>The analysis module provides the following functions: <code>runUmap()</code>, <code>calcInteger()</code>, <code>findSuggestedK()</code>, <code>findClusters()</code>, <code>calcConsensus()</code> and <code>runPhylo()</code>.</p>
<p>The visualization module functions are: <code>plotMetrics()</code>, <code>plotRatio()</code>, <code>plotUmap()</code>, <code>plotHeatmap()</code>, <code>plotConsensusLine()</code> and <code>plotGeneCopy()</code>.</p>
<div id="plotmetrics" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> plotMetrics()</h2>
<p><code>plotMetrics()</code> can be used to plot any information available in the `colData(). As a reminder, we can check the elements within coldata() with:</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="analysis-and-visualization-module.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tumor))</span></code></pre></div>
<pre><code>##  [1] &quot;sample&quot;                &quot;reads_assigned_bins&quot;   &quot;reads_unmapped&quot;       
##  [4] &quot;reads_duplicates&quot;      &quot;reads_multimapped&quot;     &quot;reads_unassigned&quot;     
##  [7] &quot;reads_ambiguous&quot;       &quot;reads_total&quot;           &quot;percentage_duplicates&quot;
## [10] &quot;overdispersion&quot;        &quot;breakpoint_count&quot;      &quot;is_normal&quot;            
## [13] &quot;find_normal_cv&quot;        &quot;filter_corr_value&quot;     &quot;filtered&quot;</code></pre>
<p><code>plotMetrics()</code> argument <code>metric</code> defines which information to plot. whereas the argument label, if supplied, will color the plot based on that information.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="analysis-and-visualization-module.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMetrics</span>(tumor, <span class="at">metric =</span> <span class="fu">c</span>(<span class="st">&quot;overdispersion&quot;</span>, </span>
<span id="cb57-2"><a href="analysis-and-visualization-module.html#cb57-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;breakpoint_count&quot;</span>,</span>
<span id="cb57-3"><a href="analysis-and-visualization-module.html#cb57-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_total&quot;</span>,</span>
<span id="cb57-4"><a href="analysis-and-visualization-module.html#cb57-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_duplicates&quot;</span>,</span>
<span id="cb57-5"><a href="analysis-and-visualization-module.html#cb57-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_assigned_bins&quot;</span>,</span>
<span id="cb57-6"><a href="analysis-and-visualization-module.html#cb57-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;percentage_duplicates&quot;</span>),</span>
<span id="cb57-7"><a href="analysis-and-visualization-module.html#cb57-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="st">&quot;overdispersion&quot;</span>)</span></code></pre></div>
<pre><code>## Coloring by: overdispersion</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="plotratio" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> plotRatio()</h2>
<p>It is often important to visualize that the segmentation behaved as expected and closely follows the ratios. Not missing important events. Ratio plots are a great tool for this. You can plot ratio plots with the <code>plotRatio()</code> function. <code>plotRatio()</code> has two different modes. When the input is the CopyKit object an interactive app will run and you can select which cell will be visualized.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="analysis-and-visualization-module.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_ratio_shiny.png" title="fig:" alt="Ratio Plot App" />
</center>
<p>If <code>plotRatio()</code> is given a sample name only the plot for that cell will be shown.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="analysis-and-visualization-module.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_sample-1.png" width="672" /></p>
</div>
<div id="runumap" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> runUmap()</h2>
<p>runUmap uses generates a <a href="https://umap-learn.readthedocs.io/en/latest/">UMAP</a> reduced dimensions embedding, the resulting embedding is stored within the scCNA reducedDim slot and is an important pre-processing step to the findClusters feature.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="analysis-and-visualization-module.html#cb61-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: SingleCellExperiment::reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<p>Additional arguments to control umap parameters can be passed on to<code>runUmap()</code> with the ‘…’ argument.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="analysis-and-visualization-module.html#cb66-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(tumor, <span class="at">n_neighbors =</span> <span class="dv">30</span>, <span class="at">min_dist =</span> <span class="dv">0</span>)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: SingleCellExperiment::reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
</div>
<div id="plotumap" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> plotUmap()</h2>
<p><code>plotUmap()</code> can be used to plot the reduced dimensional embedding. plotUmap can be colored by any element of the metadata with the argument ‘label,’ this will be shown in a later section.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="analysis-and-visualization-module.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="clustering" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Clustering</h2>
<p>findClusters uses the reduced dimensional embedding resulting from <code>runUmap()</code> to perform clustering at two levels, hereby referred to as superclones and subclones. When clustering for superclones, <code>findClusters()</code> creates a graph representation of the reduced dimension embedding using a shared nearest neighbor algorithm (SNN), from this graph the connected components are extracted and generally represent high-level structures that share large, lineage defining copy number events. At a more fine-grained resolution, CopyKit can also be used to detect subclones, i. e. groups of cells containing more exclusive copy number event per cluster. To do so, the umap embedding is again used as the pre-processing step, this time to perform a density-based clustering with hdbscan hdbscan.</p>
<p>The <code>findClusters()</code> function will use the reduced dimensional embedding resulting from <code>runUmap()</code> to perform clustering of superclones and subclones.</p>
<p>When clustering for superclones <code>findClusters()</code> creates a graph representation of the dataset reduced dimension embedding using a shared nearest neighbor algorithm (SNN), from this graph the connected components are extracted and generally represent high-level structures that share large, lineage defining copy number events.</p>
<p>CopyKit can also be used to detect subclones, i. e. groups of cells containing a unique copy number event per cluster, to do so the umap embedding is again used as the pre-processing step, this time to perform a density-based clustering with <a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">hdbscan</a> previously applied in different copy number analysis, as seen, in the work from Laks <em>et al.</em> <span class="citation">(<strong>RN6?</strong>)</span>.</p>
<p>Network clustering algorithms can also be used to detect subclones. By setting the argument <code>method</code> to <em>leiden</em> CopyKit will again use the SNN representation of the reduced dimensional embedding to apply the graph based clustering.</p>
<div id="findsugestedk" class="section level3" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> findSugestedK()</h3>
<p>Density-based clustering requires the input of the k neighbors, frequently chosen ad hoc. To help with parametrization, CopyKit provides the helper findSuggestedK function. findSuggestedK performs a clustering bootstrapping over a range of k values and returns the value that maximizes the jaccard similarity, by default the metric being maximized is <em>mean</em>, however, it is also possible to maximize the <em>median</em> with the argument <code>metric</code>. While findSuggestedK does not guarantee that the suggested value will result in optimal clustering but provides a guide that maximizes cluster stability.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="analysis-and-visualization-module.html#cb73-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(tumor)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 15 with mean jaccard similarity of: 0.992</code></pre>
<p>CopyKit provides different ways to inspect the results from <code>findSuggestedK()</code> with <code>plotSuggestedK()</code>. The default geom plots a boxplot showing the mean values of each k assessed during the grid search as the red point inside each boxplot</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="analysis-and-visualization-module.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_boxplot-1.png" width="672" /></p>
<p>if the argument <code>geom</code> is set to <em>tile</em>, <code>plotSuggestedK()</code> plots a heatmap where each row is a detected subclone, each column is a k assessed during the grid search and the color represents the jaccard similarity for a given clone. Clones not detected in a given k are presented in the color gray.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="analysis-and-visualization-module.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;tile&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_tile-1.png" width="672" /></p>
<p>It is also possible to visualize the relation between subclone sizes with the jaccard similarity setting geom to <em>scatterplot</em></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="analysis-and-visualization-module.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;scatterplot&#39;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_scatter-1.png" width="576" /></p>
<p>The suggested value is stored into the metadata and can be accessed at:</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="analysis-and-visualization-module.html#cb81-1" aria-hidden="true" tabindex="-1"></a>S4Vectors<span class="sc">::</span><span class="fu">metadata</span>(tumor)<span class="sc">$</span>suggestedK</span></code></pre></div>
<pre><code>## [1] 15</code></pre>
</div>
<div id="findclusters" class="section level3" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> findClusters()</h3>
<p>To run <code>findClusters()</code> simply use the function in the CopyKit object and provide the k value with the arguments ‘k_superclones’ and ‘k_subclones.’
<em>NOTE:</em> k_superclones and k_subclones can be used concurrently. Only k_subclones is a mandatory field.</p>
<p>By default, <code>findClusters()</code> will use the <a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">hdbscan</a> method for clustering</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="analysis-and-visualization-module.html#cb83-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor,</span>
<span id="cb83-2"><a href="analysis-and-visualization-module.html#cb83-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_superclones =</span> <span class="dv">30</span>,</span>
<span id="cb83-3"><a href="analysis-and-visualization-module.html#cb83-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_subclones =</span> <span class="dv">15</span>)</span></code></pre></div>
<p>If <code>findSuggestedK()</code> was used and the argument ‘k_subclones’ is not provided, <code>findClusters()</code> will automatically use the value resulting from <code>findSuggestedK()</code> that was stored into the metadata:</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="analysis-and-visualization-module.html#cb84-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 15</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 4 subclones.</code></pre>
<pre><code>## 0 cells were classified as outliers. Check subclone group &#39;c0&#39;.</code></pre>
<pre><code>## Done.</code></pre>
<p>If the method used for clustering was <em>hdbscan</em>, it is possible that a subgroup of outliers is identified, those are added to subgroup <code>c0</code> and may be removed by subsetting the copykit object. Copykit will inform if any cell was classified as c0 in an informative message after running <code>findClusters()</code></p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="analysis-and-visualization-module.html#cb90-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> tumor[,<span class="fu">colData</span>(tumor)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span></code></pre></div>
<p>The results from the clustering can be visualized with different plotting functions. For example <code>plotUmap()</code> can be used with the argument ‘label’:</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="analysis-and-visualization-module.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_umap_subclones-1.png" width="672" /></p>
</div>
</div>
<div id="runphylo" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> runPhylo()</h2>
<p>CopyKit can store phylogenetic trees inside the scCNA object. To run a phylogenetic tree simply use the function <code>runPhylo()</code>. The resulting tree is stored within the scCNA object in the phylo slot:</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="analysis-and-visualization-module.html#cb94-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runPhylo</span>(tumor, <span class="at">metric =</span> <span class="st">&#39;manhattan&#39;</span>)</span></code></pre></div>
<pre><code>## Using ratio data...</code></pre>
<pre><code>## Calculating distance matrix</code></pre>
<pre><code>## Creating neighbor-joining tree.</code></pre>
<pre><code>## Access slot with copykit::phylo(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
</div>
<div id="plotphylo" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> plotPhylo()</h2>
<p>Resulting phylogenetic trees from <code>runPhylo()</code> can be plotted with the function <code>plotPhylo()</code></p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="analysis-and-visualization-module.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo-1.png" width="672" /></p>
<p><code>plotPhylo()</code> can use any element of the metadata to color the leaves of the tree</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="analysis-and-visualization-module.html#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo_label-1.png" width="672" /></p>
</div>
<div id="calcinteger" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> calcInteger()</h2>
<p>Segment mean ratios can be scaled to integer values, this process of ploidy scaling is accomplished within CopyKit with <code>calcInteger()</code>. CopyKit supports different methods of calculating integer copy number profiles.</p>
<p>To calculate computational ploidies CopyKit adopts the <a href="">scquantum</a> package. To use scquantum just set the argument method to <em>scquantum</em>.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="analysis-and-visualization-module.html#cb102-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">method =</span> <span class="st">&#39;scquantum&#39;</span>)</span></code></pre></div>
<p>By setting the argument <code>method</code> to <em>fixed</em>, a fixed value of ploidy (generally determined using Flow Cytometry) will scale all cells.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="analysis-and-visualization-module.html#cb103-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">ploidy_value =</span> <span class="fl">4.26</span>)</span></code></pre></div>
<pre><code>## Scaling ratio values by ploidy value 4.26</code></pre>
<p>Alternatively, if different ploidies are needed for different cells, a vector containing the ploidies can be passed on to <code>calcInteger()</code> as long as it contains the same length and sample order as <code>colData()</code></p>
<p>The integer segment mean ratios are stored in the slot <code>integer</code> that can be accessed with the function <code>assay()</code></p>
<p>If a value of ploidy exists, <code>plotRatio()</code> will use it to plot the integer copy number values as a secondary axis.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="analysis-and-visualization-module.html#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_integer-1.png" width="672" /></p>
</div>
<div id="plotheatmap" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> plotHeatmap()</h2>
<p>A heatmap can be used to visualize the copy number profiles of the dataset annotated with the subclonal information:</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="analysis-and-visualization-module.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p>Integer copy number heatmaps can be plotted with <code>plotHeatmap()</code> and the argument <code>assay = 'integer'</code>, importantly the integer matrix must be in the <code>'integer'</code> slot.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="analysis-and-visualization-module.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p>Importantly, new information can be added to the metadata and used in conjunction with the plotting functions. The dataset in this example has some spatial information obtained from the macrodissection of the tissue. It is encoded in the sample name by the letter S followed by a number. We can extract that information and add it as an extra column to the metadata:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="analysis-and-visualization-module.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tumor)<span class="sc">$</span>spatial_info <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">colData</span>(tumor)<span class="sc">$</span>sample, <span class="st">&quot;L[0-9]&quot;</span>)</span></code></pre></div>
<p>Once the information has been added, we can use it to color the heatmap points by their spatial information:</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="analysis-and-visualization-module.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: spatial_info.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/umap_spatial-1.png" width="672" /></p>
<p>It is also possible to annotate the heatmap with that information:
The ‘label’ argument for <code>plotHeatmap()</code> can add as many annotations as specified by the user as long as they are elements in <code>colData()</code> of the CopyKit object.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="analysis-and-visualization-module.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;spatial_info&quot;</span>, <span class="st">&quot;subclones&quot;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p>If <code>calcInteger()</code> was performed, it is possible to visualize how much the copy number calls deviate from the expected integer states by using <code>plotHeatmap()</code> with the argument <code>rounding_error</code> set to <code>TRUE</code></p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="analysis-and-visualization-module.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>, <span class="at">rounding_error =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
</div>
<div id="calcconsensus" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> calcConsensus()</h2>
<p>Consensus sequences can help visualize the different segments across the groups of cells. A consensus sequence can be calculated in CopyKit with the function <code>calcConsensus()</code></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="analysis-and-visualization-module.html#cb118-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor)</span></code></pre></div>
<p><code>plotHeatmap()</code> can also plot consensus heatmap of the CopyKit object:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="analysis-and-visualization-module.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<pre><code>## Warning: The input is a data frame, convert it to the matrix.</code></pre>
<p>By default <code>calcConsensus()</code> uses the subclones information to calculate a consensus for each subclone. Importantly, any element of the <code>colData()</code> can be used to calculate the consensus.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="analysis-and-visualization-module.html#cb122-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor, <span class="at">consensus_by =</span> <span class="st">&quot;spatial_info&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="analysis-and-visualization-module.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<pre><code>## Warning: The input is a data frame, convert it to the matrix.</code></pre>
<p><strong>Note:</strong> Consensus matrices can be calculated from the integer assay. Importantly, the integer matrix must be in the <code>assay(scCNA, 'integer')</code> slot. Check <code>calcInteger()</code> for more info.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="analysis-and-visualization-module.html#cb126-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor, <span class="at">consensus_by =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
</div>
<div id="plotconsensusline" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> plotConsensusLine()</h2>
<p>To make it easier to compare the differences among subclones, <code>plotConsensusLine()</code> opens an app where the consensus sequences are plotted as lines and a checkbox can be used to add overlays of the sequences. Making it easier to determine the differences between regions.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="analysis-and-visualization-module.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotConsensusLine</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_consensus_shiny.png" title="fig:" alt="Plot Consensus Line" />
</center>
</div>
<div id="plotgenecopy" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> plotGeneCopy()</h2>
<p>It might be of interest to check for the copy number states across all the samples for different genes. That can be done with <code>plotGeneCopy()</code> in two different geoms: “swarm” (default) or “violin.” As with other plot functions, points can be colored with the argument ‘label.’</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="analysis-and-visualization-module.html#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb128-2"><a href="analysis-and-visualization-module.html#cb128-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb128-3"><a href="analysis-and-visualization-module.html#cb128-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb128-4"><a href="analysis-and-visualization-module.html#cb128-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb128-5"><a href="analysis-and-visualization-module.html#cb128-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb128-6"><a href="analysis-and-visualization-module.html#cb128-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb128-7"><a href="analysis-and-visualization-module.html#cb128-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb128-8"><a href="analysis-and-visualization-module.html#cb128-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb128-9"><a href="analysis-and-visualization-module.html#cb128-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb128-10"><a href="analysis-and-visualization-module.html#cb128-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb128-11"><a href="analysis-and-visualization-module.html#cb128-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb128-12"><a href="analysis-and-visualization-module.html#cb128-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Coloring by: spatial_info</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>It is also possible to visualize the integer data as a frequency barplot for each gene:</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="analysis-and-visualization-module.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb130-2"><a href="analysis-and-visualization-module.html#cb130-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb130-3"><a href="analysis-and-visualization-module.html#cb130-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb130-4"><a href="analysis-and-visualization-module.html#cb130-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb130-5"><a href="analysis-and-visualization-module.html#cb130-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb130-6"><a href="analysis-and-visualization-module.html#cb130-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb130-7"><a href="analysis-and-visualization-module.html#cb130-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb130-8"><a href="analysis-and-visualization-module.html#cb130-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb130-9"><a href="analysis-and-visualization-module.html#cb130-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb130-10"><a href="analysis-and-visualization-module.html#cb130-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb130-11"><a href="analysis-and-visualization-module.html#cb130-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb130-12"><a href="analysis-and-visualization-module.html#cb130-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom =</span> <span class="st">&#39;barplot&#39;</span>,</span>
<span id="cb130-13"><a href="analysis-and-visualization-module.html#cb130-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_gene_copy_integer-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="qc-module.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
