<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Analysis and Visualization module | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.23 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Analysis and Visualization module | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="qc-module.html"/>
<link rel="next" href="session-info.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1</b> runVarbin()</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#cbs-dnacopy"><i class="fa fa-check"></i><b>4.1.1</b> CBS {DNACopy}</a></li>
<li class="chapter" data-level="4.1.2" data-path="pre-processing.html"><a href="pre-processing.html#multipcf-copynumber"><i class="fa fa-check"></i><b>4.1.2</b> multipcf {copynumber}</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findnormalcells"><i class="fa fa-check"></i><b>5.2</b> findNormalCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#filtercells"><i class="fa fa-check"></i><b>5.3</b> filterCells()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>6</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>6.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>6.2</b> plotRatio()</a></li>
<li class="chapter" data-level="6.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>6.3</b> runUmap()</a></li>
<li class="chapter" data-level="6.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>6.4</b> plotUmap()</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>6.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>6.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="6.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>6.5.2</b> findClusters()</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo"><i class="fa fa-check"></i><b>6.6</b> runPhylo()</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotphylo"><i class="fa fa-check"></i><b>6.7</b> plotPhylo()</a></li>
<li class="chapter" data-level="6.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcinteger"><i class="fa fa-check"></i><b>6.8</b> calcInteger()</a></li>
<li class="chapter" data-level="6.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>6.9</b> plotHeatmap()</a></li>
<li class="chapter" data-level="6.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>6.10</b> plotFreq()</a></li>
<li class="chapter" data-level="6.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>6.11</b> calcConsensus()</a></li>
<li class="chapter" data-level="6.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>6.12</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="6.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findvariablegenes"><i class="fa fa-check"></i><b>6.13</b> findVariableGenes()</a></li>
<li class="chapter" data-level="6.14" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotvariablegenes"><i class="fa fa-check"></i><b>6.14</b> plotVariableGenes()</a></li>
<li class="chapter" data-level="6.15" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>6.15</b> plotGeneCopy()</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>7</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis-and-visualization-module" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Analysis and Visualization module</h1>
<p>The analysis and visualization module from CopyKit work in synergy to help you analyze and assess the results from your day.</p>
<p>The analysis module provides the following functions: <code>runUmap()</code>, <code>calcInteger()</code>, <code>findSuggestedK()</code>, <code>findClusters()</code>, <code>calcConsensus()</code> and <code>runPhylo()</code>.</p>
<p>The visualization module functions are: <code>plotMetrics()</code>, <code>plotRatio()</code>, <code>plotUmap()</code>, <code>plotHeatmap()</code>, <code>plotConsensusLine()</code> and <code>plotGeneCopy()</code>.</p>
<div id="plotmetrics" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> plotMetrics()</h2>
<p><code>plotMetrics()</code> can be used to plot any information available in the `colData(). As a reminder, we can check the elements within coldata() with:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="analysis-and-visualization-module.html#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tumor))</span></code></pre></div>
<pre><code>##  [1] &quot;sample&quot;                &quot;reads_assigned_bins&quot;   &quot;reads_unmapped&quot;       
##  [4] &quot;reads_duplicates&quot;      &quot;reads_multimapped&quot;     &quot;reads_unassigned&quot;     
##  [7] &quot;reads_ambiguous&quot;       &quot;reads_total&quot;           &quot;percentage_duplicates&quot;
## [10] &quot;overdispersion&quot;        &quot;breakpoint_count&quot;      &quot;is_normal&quot;            
## [13] &quot;find_normal_cv&quot;        &quot;filter_corr_value&quot;     &quot;filtered&quot;</code></pre>
<p><code>plotMetrics()</code> argument <code>metric</code> defines which information to plot. whereas the argument label, if supplied, will color the plot based on that information.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="analysis-and-visualization-module.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMetrics</span>(tumor, <span class="at">metric =</span> <span class="fu">c</span>(<span class="st">&quot;overdispersion&quot;</span>, </span>
<span id="cb61-2"><a href="analysis-and-visualization-module.html#cb61-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;breakpoint_count&quot;</span>,</span>
<span id="cb61-3"><a href="analysis-and-visualization-module.html#cb61-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_total&quot;</span>,</span>
<span id="cb61-4"><a href="analysis-and-visualization-module.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_duplicates&quot;</span>,</span>
<span id="cb61-5"><a href="analysis-and-visualization-module.html#cb61-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;reads_assigned_bins&quot;</span>,</span>
<span id="cb61-6"><a href="analysis-and-visualization-module.html#cb61-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;percentage_duplicates&quot;</span>),</span>
<span id="cb61-7"><a href="analysis-and-visualization-module.html#cb61-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">label =</span> <span class="st">&quot;overdispersion&quot;</span>)</span></code></pre></div>
<pre><code>## Coloring by: overdispersion</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="plotratio" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> plotRatio()</h2>
<p>It is often important to visualize that the segmentation behaved as expected and closely follows the ratios. Not missing important events. Ratio plots are a great tool for this. You can plot ratio plots with the <code>plotRatio()</code> function. <code>plotRatio()</code> has two different modes. When the input is the CopyKit object an interactive app will run and you can select which cell will be visualized.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="analysis-and-visualization-module.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_ratio_shiny.png" title="fig:" alt="Ratio Plot App" />
</center>
<p>If <code>plotRatio()</code> is given a sample name only the plot for that cell will be shown.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="analysis-and-visualization-module.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_sample-1.png" width="672" /></p>
</div>
<div id="runumap" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> runUmap()</h2>
<p>runUmap uses generates a <a href="https://umap-learn.readthedocs.io/en/latest/">UMAP</a> reduced dimensions embedding, the resulting embedding is stored within the scCNA reducedDim slot and is an important pre-processing step to the findClusters feature.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="analysis-and-visualization-module.html#cb65-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: SingleCellExperiment::reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
<p>Additional arguments to control umap parameters can be passed on to<code>runUmap()</code> with the ‘…’ argument. As exemplified in the code below.
By default, <em>CopyKit</em> will pass to the umap different values of n_neighbor and min_dist using: <code>n_neighbor = 30</code> and <code>min_dist = 0</code>.</p>
<p>The full list of additional arguments that can be passed on to <code>uwot::umap</code> with the ‘…’ argument to <code>runUmap()</code> can be seen in the <a href="https://cran.r-project.org/web/packages/uwot/uwot.pdf">uwot manual</a>.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="analysis-and-visualization-module.html#cb70-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runUmap</span>(tumor, <span class="at">n_neighbors =</span> <span class="dv">30</span>, <span class="at">min_dist =</span> <span class="dv">0</span>)</span></code></pre></div>
<pre><code>## Using assay: logr</code></pre>
<pre><code>## Embedding data with UMAP. Using seed 17</code></pre>
<pre><code>## Access reduced dimensions slot with: SingleCellExperiment::reducedDim(scCNA, &#39;umap&#39;).</code></pre>
<pre><code>## Done.</code></pre>
</div>
<div id="plotumap" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> plotUmap()</h2>
<p><code>plotUmap()</code> can be used to plot the reduced dimensional embedding. plotUmap can be colored by any element of the metadata with the argument ‘label,’ this will be shown in a later section.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="analysis-and-visualization-module.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="clustering" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Clustering</h2>
<p>findClusters uses the reduced dimensional embedding resulting from <code>runUmap()</code> to perform clustering at two levels, hereby referred to as superclones and subclones. When clustering for superclones, <code>findClusters()</code> creates a graph representation of the reduced dimension embedding using a shared nearest neighbor algorithm (SNN), from this graph the connected components are extracted and generally represent high-level structures that share large, lineage defining copy number events. At a more fine-grained resolution, CopyKit can also be used to detect subclones, i. e. groups of cells containing more exclusive copy number event per cluster. To do so, the umap embedding is again used as the pre-processing step, this time to perform a density-based clustering with hdbscan hdbscan.</p>
<p>The <code>findClusters()</code> function will use the reduced dimensional embedding resulting from <code>runUmap()</code> to perform clustering of superclones and subclones.</p>
<p>When clustering for superclones <code>findClusters()</code> creates a graph representation of the dataset reduced dimension embedding using a shared nearest neighbor algorithm (SNN), from this graph the connected components are extracted and generally represent high-level structures that share large, lineage defining copy number events.</p>
<p>CopyKit can also be used to detect subclones, i. e. groups of cells containing a unique copy number event per cluster, to do so the umap embedding is again used as the pre-processing step, this time to perform a density-based clustering with <a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">hdbscan</a> previously applied in different copy number analysis, as seen, in the work from Laks <em>et al.</em> <span class="citation">(<strong>RN6?</strong>)</span>.</p>
<p>Network clustering algorithms can also be used to detect subclones. By setting the argument <code>method</code> to <em>leiden</em> CopyKit will again use the SNN representation of the reduced dimensional embedding to apply the graph based clustering.</p>
<div id="findsugestedk" class="section level3" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> findSugestedK()</h3>
<p>Density-based clustering requires the input of the k neighbors, frequently chosen ad hoc. To help with parametrization, CopyKit provides the helper findSuggestedK function. findSuggestedK performs a clustering bootstrapping over a range of k values and returns the value that maximizes the jaccard similarity, by default the metric being maximized is <em>mean</em>, however, it is also possible to maximize the <em>median</em> with the argument <code>metric</code>. While findSuggestedK does not guarantee that the suggested value will result in optimal clustering but provides a guide that maximizes cluster stability.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="analysis-and-visualization-module.html#cb77-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findSuggestedK</span>(tumor)</span></code></pre></div>
<pre><code>## Calculating jaccard similarity for k range: 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28</code></pre>
<pre><code>## </code></pre>
<pre><code>## Suggested k = 14 with mean jaccard similarity of: 0.984</code></pre>
<p>CopyKit provides different ways to inspect the results from <code>findSuggestedK()</code> with <code>plotSuggestedK()</code>. The default geom plots a boxplot showing the mean values of each k assessed during the grid search as the red point inside each boxplot</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="analysis-and-visualization-module.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_boxplot-1.png" width="672" /></p>
<p>if the argument <code>geom</code> is set to <em>tile</em>, <code>plotSuggestedK()</code> plots a heatmap where each row is a detected subclone, each column is a k assessed during the grid search and the color represents the jaccard similarity for a given clone. Clones not detected in a given k are presented in the color gray.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="analysis-and-visualization-module.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;tile&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_tile-1.png" width="672" /></p>
<p>It is also possible to visualize the relation between subclone sizes with the jaccard similarity setting geom to <em>scatterplot</em></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="analysis-and-visualization-module.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSuggestedK</span>(tumor, <span class="at">geom =</span> <span class="st">&#39;scatterplot&#39;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_suggested_k_scatter-1.png" width="576" /></p>
<p>The suggested value is stored into the metadata and can be accessed at:</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="analysis-and-visualization-module.html#cb85-1" aria-hidden="true" tabindex="-1"></a>S4Vectors<span class="sc">::</span><span class="fu">metadata</span>(tumor)<span class="sc">$</span>suggestedK</span></code></pre></div>
<pre><code>## [1] 14</code></pre>
</div>
<div id="findclusters" class="section level3" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> findClusters()</h3>
<p>To run <code>findClusters()</code> simply use the function in the CopyKit object and provide the k value with the arguments ‘k_superclones’ and ‘k_subclones.’
<em>NOTE:</em> k_superclones and k_subclones can be used concurrently. Only k_subclones is a mandatory field.</p>
<p>By default, <code>findClusters()</code> will use the <a href="https://hdbscan.readthedocs.io/en/latest/how_hdbscan_works.html">hdbscan</a> method for clustering</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="analysis-and-visualization-module.html#cb87-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor,</span>
<span id="cb87-2"><a href="analysis-and-visualization-module.html#cb87-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_superclones =</span> <span class="dv">30</span>,</span>
<span id="cb87-3"><a href="analysis-and-visualization-module.html#cb87-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">k_subclones =</span> <span class="dv">15</span>)</span></code></pre></div>
<p>If <code>findSuggestedK()</code> was used and the argument ‘k_subclones’ is not provided, <code>findClusters()</code> will automatically use the value resulting from <code>findSuggestedK()</code> that was stored into the metadata:</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="analysis-and-visualization-module.html#cb88-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findClusters</span>(tumor)</span></code></pre></div>
<pre><code>## Using suggested k_subclones = 14</code></pre>
<pre><code>## Finding clusters, using method: hdbscan</code></pre>
<pre><code>## Found 4 subclones.</code></pre>
<pre><code>## Done.</code></pre>
<p>If the method used for clustering was <em>hdbscan</em>, it is possible that a subgroup of outliers is identified, those are added to subgroup <code>c0</code> and may be removed by subsetting the copykit object. Copykit will inform if any cell was classified as c0 in an informative message after running <code>findClusters()</code></p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="analysis-and-visualization-module.html#cb93-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> tumor[,<span class="fu">colData</span>(tumor)<span class="sc">$</span>subclones <span class="sc">!=</span> <span class="st">&#39;c0&#39;</span>]</span></code></pre></div>
<p>The results from the clustering can be visualized with different plotting functions. For example <code>plotUmap()</code> can be used with the argument ‘label’:</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="analysis-and-visualization-module.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: subclones.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_umap_subclones-1.png" width="672" /></p>
</div>
</div>
<div id="runphylo" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> runPhylo()</h2>
<p>CopyKit can store phylogenetic trees inside the scCNA object. To run a phylogenetic tree simply use the function <code>runPhylo()</code>. The resulting tree is stored within the scCNA object in the phylo slot:</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="analysis-and-visualization-module.html#cb97-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runPhylo</span>(tumor, <span class="at">metric =</span> <span class="st">&#39;manhattan&#39;</span>)</span></code></pre></div>
<pre><code>## Using ratio data...</code></pre>
<pre><code>## Calculating distance matrix</code></pre>
<pre><code>## Creating neighbor-joining tree.</code></pre>
<pre><code>## Access slot with copykit::phylo(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
</div>
<div id="plotphylo" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> plotPhylo()</h2>
<p>Resulting phylogenetic trees from <code>runPhylo()</code> can be plotted with the function <code>plotPhylo()</code></p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="analysis-and-visualization-module.html#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo-1.png" width="672" /></p>
<p><code>plotPhylo()</code> can use any element of the metadata to color the leaves of the tree</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="analysis-and-visualization-module.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPhylo</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_phylo_label-1.png" width="672" /></p>
</div>
<div id="calcinteger" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> calcInteger()</h2>
<p>Segment mean ratios can be scaled to integer values, this process of ploidy scaling is accomplished within CopyKit with <code>calcInteger()</code>. CopyKit supports different methods of calculating integer copy number profiles.</p>
<p>To calculate computational ploidies CopyKit adopts the <a href="">scquantum</a> package. To use scquantum just set the argument method to <em>scquantum</em>.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="analysis-and-visualization-module.html#cb105-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">method =</span> <span class="st">&#39;scquantum&#39;</span>)</span></code></pre></div>
<p>By setting the argument <code>method</code> to <em>fixed</em>, a fixed value of ploidy (generally determined using Flow Cytometry) will scale all cells.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="analysis-and-visualization-module.html#cb106-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcInteger</span>(tumor, <span class="at">ploidy_value =</span> <span class="fl">4.3</span>)</span></code></pre></div>
<pre><code>## Scaling ratio values by ploidy value 4.3</code></pre>
<p>Alternatively, if different ploidies are needed for different cells, a vector containing the ploidies can be passed on to <code>calcInteger()</code> as long as it contains the same length and sample order as <code>colData()</code></p>
<p>The integer segment mean ratios are stored in the slot <code>integer</code> that can be accessed with the function <code>assay()</code></p>
<p>If a value of ploidy exists, <code>plotRatio()</code> will use it to plot the integer copy number values as a secondary axis.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="analysis-and-visualization-module.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRatio</span>(tumor, <span class="st">&quot;PMTC6LiverC117AL4L5S1_S885_L003_R1_001&quot;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ratio_integer-1.png" width="672" /></p>
</div>
<div id="plotheatmap" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> plotHeatmap()</h2>
<p>A heatmap can be used to visualize the copy number profiles of the dataset annotated with the subclonal information:</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="analysis-and-visualization-module.html#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_heatmap_subclones-1.png" width="672" /></p>
<p>Integer copy number heatmaps can be plotted with <code>plotHeatmap()</code> and the argument <code>assay = 'integer'</code>, importantly the integer matrix must be in the <code>'integer'</code> slot.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="analysis-and-visualization-module.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_ht_int-1.png" width="672" /></p>
<p>Importantly, new information can be added to the metadata and used in conjunction with the plotting functions. The dataset in this example has some spatial information obtained from the macrodissection of the tissue. It is encoded in the sample name by the letter S followed by a number. We can extract that information and add it as an extra column to the metadata:</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="analysis-and-visualization-module.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tumor)<span class="sc">$</span>spatial_info <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(<span class="fu">colData</span>(tumor)<span class="sc">$</span>sample, <span class="st">&quot;L[0-9]&quot;</span>)</span></code></pre></div>
<p>Once the information has been added, we can use it to color the heatmap points by their spatial information:</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="analysis-and-visualization-module.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotUmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Umap.</code></pre>
<pre><code>## Coloring by: spatial_info.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/umap_spatial-1.png" width="672" /></p>
<p>It is also possible to annotate the heatmap with that information:
The ‘label’ argument for <code>plotHeatmap()</code> can add as many annotations as specified by the user as long as they are elements in <code>colData()</code> of the CopyKit object.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="analysis-and-visualization-module.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="fu">c</span>(<span class="st">&quot;spatial_info&quot;</span>, <span class="st">&quot;subclones&quot;</span>))</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/heat_spatial-1.png" width="672" /></p>
<p>If <code>calcInteger()</code> was performed, it is possible to visualize how much the copy number calls deviate from the expected integer states by using <code>plotHeatmap()</code> with the argument <code>rounding_error</code> set to <code>TRUE</code></p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="analysis-and-visualization-module.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>, <span class="at">rounding_error =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/rounding_error-1.png" width="672" /></p>
</div>
<div id="plotfreq" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> plotFreq()</h2>
<p>Computing the frequencies of genomic gain or losses across the genome can be useful to visualize the differences between groups. This can be done with <code>plotFreq()</code>.
For every region of the genome, <code>plotFreq()</code> will calculate the frequency of gain or losses according to a determined threshold across all samples.
The thresholds are controlled with the arguments <code>low_threshold</code> (values below this value will be counted as genomic losses) and <code>high_threshold</code> (values above this value will be counted as genomic gains).
Ideally those values will be set accordingly to the ploidy of the sample. the sample used in the example throughout this documentation has a ploidy of approximately 4.3, therefore a gain or loss of one copy number for this sample should have a difference in the segment ratio value of 1/4.3 (~0.23). Therefore we will set the accordingly to a difference of plus or minus 0.2 (leaving some room for error), compared to a segment ratio of 1.</p>
<p>The argument <code>assay</code> can be provided to pass on the <code>integer</code> assay instead of the <code>segment_ratios</code> assay.</p>
<p>Two geoms are available <code>area</code> or <code>line</code></p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="analysis-and-visualization-module.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotFreq</span>(tumor, <span class="at">low_threshold =</span> <span class="fl">0.8</span>, <span class="at">high_threshold =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_freq_sample-1.png" width="672" /></p>
<p>Group information can be set to any categorical element of the metadata and is provided with the argument <code>label</code>. If no information is provided on <code>label</code> a frequency plot of the entire sample will be plotted.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="analysis-and-visualization-module.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotFreq</span>(tumor, </span>
<span id="cb122-2"><a href="analysis-and-visualization-module.html#cb122-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">low_threshold =</span> <span class="fl">0.8</span>, </span>
<span id="cb122-3"><a href="analysis-and-visualization-module.html#cb122-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">high_threshold =</span> <span class="fl">1.2</span>, </span>
<span id="cb122-4"><a href="analysis-and-visualization-module.html#cb122-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_freq_label-1.png" width="672" /></p>
</div>
<div id="calcconsensus" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> calcConsensus()</h2>
<p>Consensus sequences can help visualize the different segments across the groups of cells. A consensus sequence can be calculated in CopyKit with the function <code>calcConsensus()</code></p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="analysis-and-visualization-module.html#cb123-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor)</span></code></pre></div>
<p><code>plotHeatmap()</code> can also plot consensus heatmap of the CopyKit object:</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="analysis-and-visualization-module.html#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;subclones&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_heat_consensus-1.png" width="672" /></p>
<p>By default <code>calcConsensus()</code> uses the subclones information to calculate a consensus for each subclone. Importantly, any element of the <code>colData()</code> can be used to calculate the consensus.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="analysis-and-visualization-module.html#cb126-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor, <span class="at">consensus_by =</span> <span class="st">&quot;spatial_info&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="analysis-and-visualization-module.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">consensus =</span> <span class="cn">TRUE</span>, <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/consensus_spatial_head-1.png" width="672" /></p>
<p><strong>Note:</strong> Consensus matrices can be calculated from the integer assay. Importantly, the integer matrix must be in the <code>assay(scCNA, 'integer')</code> slot. Check <code>calcInteger()</code> for more info.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="analysis-and-visualization-module.html#cb129-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">calcConsensus</span>(tumor, <span class="at">consensus_by =</span> <span class="st">&#39;subclones&#39;</span>, <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
</div>
<div id="plotconsensusline" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> plotConsensusLine()</h2>
<p>To make it easier to compare the differences among subclones, <code>plotConsensusLine()</code> opens an app where the consensus sequences are plotted as lines and a checkbox can be used to add overlays of the sequences. Making it easier to determine the differences between regions.</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="analysis-and-visualization-module.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotConsensusLine</span>(tumor)</span></code></pre></div>
<center>
<img src="images/plot_consensus_shiny.png" title="fig:" alt="Plot Consensus Line" />
</center>
</div>
<div id="findvariablegenes" class="section level2" number="6.13">
<h2><span class="header-section-number">6.13</span> findVariableGenes()</h2>
<p>Finding variable genes can provide important information as to which genes contribute the most to the subclonal structure of the sample. This can be done with <code>findVariableGenes()</code>.</p>
<p><code>findVariableGenes()</code> requires a list of genes to be provided with the argument <code>genes</code>. To facilitate <em>CopyKit</em> provides a list of 100 putative oncogenes and tumor supressor genes from <a href="https://www.sciencedirect.com/science/article/pii/S0092867413012877?via%3Dihub">Davoli et al.</a> in a vector called <code>tumor_genes</code> which is loaded with CopyKit.
The argument <code>top_n</code> controls the number of genes to be returned</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="analysis-and-visualization-module.html#cb131-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findVariableGenes</span>(tumor, </span>
<span id="cb131-2"><a href="analysis-and-visualization-module.html#cb131-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">genes =</span> tumor_genes,</span>
<span id="cb131-3"><a href="analysis-and-visualization-module.html#cb131-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">top_n =</span> <span class="dv">30</span>)</span></code></pre></div>
<pre><code>## Warning in findVariableGenes(tumor, genes = tumor_genes, top_n = 30): Genes: H3F3A,
## MYCL1 ,could not be found. Maybe you need to use a different gene alias?</code></pre>
<pre><code>## Warning in findVariableGenes(tumor, genes = tumor_genes, top_n = 30): Genes: BCL6, HRAS,
## SMAD2 are in blacklisted regions of the Varbin pipeline and can&#39;t be plotted.</code></pre>
<p>The resulting list of variable genes is stored within the metadata of the CopyKit object.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="analysis-and-visualization-module.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(tumor)<span class="sc">$</span>hvg</span></code></pre></div>
<pre><code>##  [1] &quot;SPOP&quot;   &quot;COL1A1&quot; &quot;ARID2&quot;  &quot;KIT&quot;    &quot;KRAS&quot;   &quot;CDK4&quot;   &quot;CDH1&quot;   &quot;IL7R&quot;   &quot;PDGFRA&quot;
## [10] &quot;BRCA2&quot;  &quot;RB1&quot;    &quot;WT1&quot;    &quot;RUNX1&quot;  &quot;ERG&quot;    &quot;MED12&quot;  &quot;MDM2&quot;   &quot;ATRX&quot;   &quot;MDM4&quot;  
## [19] &quot;CDKN1A&quot; &quot;FGFR1&quot;  &quot;DAXX&quot;   &quot;NRAS&quot;   &quot;U2AF1&quot;  &quot;BRCA1&quot;  &quot;ERBB2&quot;  &quot;NOTCH2&quot; &quot;MYB&quot;   
## [28] &quot;APC&quot;    &quot;PIK3CA&quot; &quot;AR&quot;    
## attr(,&quot;pca_pc1_loading&quot;)
##          CDK4        CDKN1A        CDKN1B        CDKN2A         CHEK1        COL1A1 
## -0.0146710466  0.0084660752  0.0043957841  0.0026456534 -0.0053863527 -0.4384591274 
##        CTNNB1          DAXX        DNMT3A          EGFR         ARID2        EIF4A2 
## -0.0029141386  0.0078996898  0.0013694282  0.0030274619 -0.0235771528  0.0028023882 
##         ERBB2          AKT1           ERG          AKT2          ETV1          ETV6 
## -0.0074511420  0.0003801861 -0.0101644600  0.0022627742 -0.0021560482  0.0048660609 
##         FGFR1         FGFR3         FGFR2          FLT3         SF3B1           ALK 
## -0.0081407244 -0.0038136195 -0.0001337979  0.0038390034  0.0020168912 -0.0009974426 
##         GATA2         GATA3         SETD2          MSH6           APC          IDH1 
## -0.0011920018 -0.0035316184 -0.0003032341  0.0002113814 -0.0060846679  0.0007084374 
##          IDH2           FAS          IL7R            AR          JAK1          JAK2 
##  0.0017993149  0.0035946974  0.0129418231  0.0055880251  0.0010669074  0.0047430418 
##           JUN           KIT          KRAS         SMAD4          MDM2          MDM4 
##  0.0004731041 -0.0172508537 -0.0172106303 -0.0012651471 -0.0091080232 -0.0086123883 
##        MAP3K1          MEN1           MET          MLH1         KMT2A          MSH2 
## -0.0047596728 -0.0042184330  0.0009217338 -0.0001570824 -0.0037168347  0.0002113814 
##           MYB           MYC          MYCN         MYD88           ATM           NF1 
## -0.0066053909  0.0031013296 -0.0003195861 -0.0008913659 -0.0046481874 -0.0032873824 
##           NF2        NOTCH1        NOTCH2          NRAS        PDGFRA        PIK3CA 
## -0.0030206444  0.0013750952 -0.0069624215 -0.0078532084 -0.0120269384  0.0058221809 
##           ATR          ATRX          TET2          BCOR       PPP2R1A         FBXW7 
##  0.0027890046  0.0089562314  0.0011002944  0.0040349846  0.0006695846 -0.0028489702 
##          PTEN        ARID1B         KMT2C           RB1         CCND1        MAP2K4 
##  0.0036215605  0.0010770593  0.0014878799  0.0115272745 -0.0036230716  0.0006150178 
##       SMARCA4       SMARCB1          SOX2         BRCA1          BRAF         BRCA2 
##  0.0047877089 -0.0013007426  0.0055466873 -0.0074511420  0.0017498759  0.0118541800 
##         STK11          TP53          TSC1          TSC2         U2AF1           VHL 
##  0.0011929999  0.0024032218  0.0012141721 -0.0015389427 -0.0077895360  0.0041595643 
##           WT1         KMT2D        ARID1A         AXIN1          BAP1          SPOP 
##  0.0105175617  0.0001220848  0.0026352946 -0.0015389427 -0.0009143539 -0.8965817276 
##         CASP8         RUNX1         CCNE1         MED12          CDH1 
##  0.0024764568 -0.0101846929 -0.0025175166  0.0092907801  0.0136316509 
## attr(,&quot;pca_df&quot;)
##            gene            p1
## SPOP       SPOP -0.8965817276
## COL1A1   COL1A1 -0.4384591274
## ARID2     ARID2 -0.0235771528
## KIT         KIT -0.0172508537
## KRAS       KRAS -0.0172106303
## CDK4       CDK4 -0.0146710466
## CDH1       CDH1  0.0136316509
## IL7R       IL7R  0.0129418231
## PDGFRA   PDGFRA -0.0120269384
## BRCA2     BRCA2  0.0118541800
## RB1         RB1  0.0115272745
## WT1         WT1  0.0105175617
## RUNX1     RUNX1 -0.0101846929
## ERG         ERG -0.0101644600
## MED12     MED12  0.0092907801
## MDM2       MDM2 -0.0091080232
## ATRX       ATRX  0.0089562314
## MDM4       MDM4 -0.0086123883
## CDKN1A   CDKN1A  0.0084660752
## FGFR1     FGFR1 -0.0081407244
## DAXX       DAXX  0.0078996898
## NRAS       NRAS -0.0078532084
## U2AF1     U2AF1 -0.0077895360
## BRCA1     BRCA1 -0.0074511420
## ERBB2     ERBB2 -0.0074511420
## NOTCH2   NOTCH2 -0.0069624215
## MYB         MYB -0.0066053909
## APC         APC -0.0060846679
## PIK3CA   PIK3CA  0.0058221809
## AR           AR  0.0055880251
## SOX2       SOX2  0.0055466873
## CHEK1     CHEK1 -0.0053863527
## ETV6       ETV6  0.0048660609
## SMARCA4 SMARCA4  0.0047877089
## MAP3K1   MAP3K1 -0.0047596728
## JAK2       JAK2  0.0047430418
## ATM         ATM -0.0046481874
## CDKN1B   CDKN1B  0.0043957841
## MEN1       MEN1 -0.0042184330
## VHL         VHL  0.0041595643
## BCOR       BCOR  0.0040349846
## FLT3       FLT3  0.0038390034
## FGFR3     FGFR3 -0.0038136195
## KMT2A     KMT2A -0.0037168347
## CCND1     CCND1 -0.0036230716
## PTEN       PTEN  0.0036215605
## FAS         FAS  0.0035946974
## GATA3     GATA3 -0.0035316184
## NF1         NF1 -0.0032873824
## MYC         MYC  0.0031013296
## EGFR       EGFR  0.0030274619
## NF2         NF2 -0.0030206444
## CTNNB1   CTNNB1 -0.0029141386
## FBXW7     FBXW7 -0.0028489702
## EIF4A2   EIF4A2  0.0028023882
## ATR         ATR  0.0027890046
## CDKN2A   CDKN2A  0.0026456534
## ARID1A   ARID1A  0.0026352946
## CCNE1     CCNE1 -0.0025175166
## CASP8     CASP8  0.0024764568
## TP53       TP53  0.0024032218
## AKT2       AKT2  0.0022627742
## ETV1       ETV1 -0.0021560482
## SF3B1     SF3B1  0.0020168912
## IDH2       IDH2  0.0017993149
## BRAF       BRAF  0.0017498759
## TSC2       TSC2 -0.0015389427
## AXIN1     AXIN1 -0.0015389427
## KMT2C     KMT2C  0.0014878799
## NOTCH1   NOTCH1  0.0013750952
## DNMT3A   DNMT3A  0.0013694282
## SMARCB1 SMARCB1 -0.0013007426
## SMAD4     SMAD4 -0.0012651471
## TSC1       TSC1  0.0012141721
## STK11     STK11  0.0011929999
## GATA2     GATA2 -0.0011920018
## TET2       TET2  0.0011002944
## ARID1B   ARID1B  0.0010770593
## JAK1       JAK1  0.0010669074
## ALK         ALK -0.0009974426
## MET         MET  0.0009217338
## BAP1       BAP1 -0.0009143539
## MYD88     MYD88 -0.0008913659
## IDH1       IDH1  0.0007084374
## PPP2R1A PPP2R1A  0.0006695846
## MAP2K4   MAP2K4  0.0006150178
## JUN         JUN  0.0004731041
## AKT1       AKT1  0.0003801861
## MYCN       MYCN -0.0003195861
## SETD2     SETD2 -0.0003032341
## MSH2       MSH2  0.0002113814
## MSH6       MSH6  0.0002113814
## MLH1       MLH1 -0.0001570824
## FGFR2     FGFR2 -0.0001337979
## KMT2D     KMT2D  0.0001220848</code></pre>
<p>Results are on decreasing order from most variable to list variable. The results can be visualized with <code>plotVariableGenes()</code> and <code>plotGeneCopy()</code>.</p>
<p>Importantly, <code>findVariableGenes()</code> cannot be used as a comparison of groups but as a way of finding the most variable genes across the entire sample. Another thing to be aware is that genes that are overexpressed and present diversity of copy number states tend to be oncogenes, therefore using only <code>findVariableGenes()</code> as a discovery tool can under emphasize events in important tumor supressor genes across a sample.</p>
</div>
<div id="plotvariablegenes" class="section level2" number="6.14">
<h2><span class="header-section-number">6.14</span> plotVariableGenes()</h2>
<p><code>plotVariableGenes()</code> is a visualization for the results from <code>findVariableGenes()</code>.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="analysis-and-visualization-module.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotVariableGenes</span>(tumor)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/pvg-1.png" width="672" /></p>
</div>
<div id="plotgenecopy" class="section level2" number="6.15">
<h2><span class="header-section-number">6.15</span> plotGeneCopy()</h2>
<p>It might be of interest to check for the copy number states across all the samples for different genes. That can be done with <code>plotGeneCopy()</code> in two different geoms: “swarm” (default) or “violin.” As with other plot functions, points can be colored with the argument ‘label.’</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="analysis-and-visualization-module.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb137-2"><a href="analysis-and-visualization-module.html#cb137-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb137-3"><a href="analysis-and-visualization-module.html#cb137-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb137-4"><a href="analysis-and-visualization-module.html#cb137-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb137-5"><a href="analysis-and-visualization-module.html#cb137-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb137-6"><a href="analysis-and-visualization-module.html#cb137-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb137-7"><a href="analysis-and-visualization-module.html#cb137-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb137-8"><a href="analysis-and-visualization-module.html#cb137-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb137-9"><a href="analysis-and-visualization-module.html#cb137-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb137-10"><a href="analysis-and-visualization-module.html#cb137-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb137-11"><a href="analysis-and-visualization-module.html#cb137-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb137-12"><a href="analysis-and-visualization-module.html#cb137-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Coloring by: spatial_info</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p><code>plotGeneCopy()</code> can be specially useful to visualize the result from <code>findVariableGenes()</code></p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="analysis-and-visualization-module.html#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor,</span>
<span id="cb139-2"><a href="analysis-and-visualization-module.html#cb139-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">genes =</span> <span class="fu">metadata</span>(tumor)<span class="sc">$</span>hvg,</span>
<span id="cb139-3"><a href="analysis-and-visualization-module.html#cb139-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="st">&#39;spatial_info&#39;</span>)</span></code></pre></div>
<pre><code>## Coloring by: spatial_info</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/pgc_fvg-1.png" width="672" /></p>
<p>A barplot geom is also provided to visualize the integer data as a frequency barplot for each gene:</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="analysis-and-visualization-module.html#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeneCopy</span>(tumor, <span class="at">genes =</span> <span class="fu">c</span>(<span class="st">&quot;CDKN2A&quot;</span>,</span>
<span id="cb141-2"><a href="analysis-and-visualization-module.html#cb141-2" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;FGFR1&quot;</span>,</span>
<span id="cb141-3"><a href="analysis-and-visualization-module.html#cb141-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb141-4"><a href="analysis-and-visualization-module.html#cb141-4" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PTEN&quot;</span>,</span>
<span id="cb141-5"><a href="analysis-and-visualization-module.html#cb141-5" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MYC&quot;</span>,</span>
<span id="cb141-6"><a href="analysis-and-visualization-module.html#cb141-6" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CDKN1A&quot;</span>,</span>
<span id="cb141-7"><a href="analysis-and-visualization-module.html#cb141-7" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;MDM2&quot;</span>,</span>
<span id="cb141-8"><a href="analysis-and-visualization-module.html#cb141-8" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;AURKA&quot;</span>,</span>
<span id="cb141-9"><a href="analysis-and-visualization-module.html#cb141-9" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;PIK3CA&quot;</span>,</span>
<span id="cb141-10"><a href="analysis-and-visualization-module.html#cb141-10" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;CCND1&quot;</span>,</span>
<span id="cb141-11"><a href="analysis-and-visualization-module.html#cb141-11" aria-hidden="true" tabindex="-1"></a>                              <span class="st">&quot;KRAS&quot;</span>),</span>
<span id="cb141-12"><a href="analysis-and-visualization-module.html#cb141-12" aria-hidden="true" tabindex="-1"></a>             <span class="at">geom =</span> <span class="st">&#39;barplot&#39;</span>,</span>
<span id="cb141-13"><a href="analysis-and-visualization-module.html#cb141-13" aria-hidden="true" tabindex="-1"></a>             <span class="at">assay =</span> <span class="st">&#39;integer&#39;</span>)</span></code></pre></div>
<p><img src="CopyKit-UserGuide_files/figure-html/plot_gene_copy_integer-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="qc-module.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="session-info.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "lunr",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
