<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Pre-processing | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Pre-processing | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Pre-processing | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="quick-start.html"/>
<link rel="next" href="qc-module.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#from-single-cell-bam-files"><i class="fa fa-check"></i><b>4.1</b> From Single Cell BAM files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1.1</b> runVarbin()</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin-modules"><i class="fa fa-check"></i><b>4.2</b> runVarbin() modules</a></li>
<li class="chapter" data-level="4.3" data-path="pre-processing.html"><a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds"><i class="fa fa-check"></i><b>4.3</b> From external count data or user-defined scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findaneuploidcells"><i class="fa fa-check"></i><b>5.2</b> findAneuploidCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#findoutliers"><i class="fa fa-check"></i><b>5.3</b> findOutliers()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="knn-smoothing.html"><a href="knn-smoothing.html"><i class="fa fa-check"></i><b>6</b> kNN smoothing</a></li>
<li class="chapter" data-level="7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>7</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="7.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>7.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="7.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>7.2</b> plotRatio()</a></li>
<li class="chapter" data-level="7.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>7.3</b> runUmap()</a></li>
<li class="chapter" data-level="7.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>7.4</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>7.4.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="7.4.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>7.4.2</b> findClusters()</a></li>
<li class="chapter" data-level="7.4.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>7.4.3</b> plotUmap()</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#scquantum-calcinteger"><i class="fa fa-check"></i><b>7.5</b> scquantum &amp; calcInteger()</a></li>
<li class="chapter" data-level="7.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>7.6</b> plotHeatmap()</a></li>
<li class="chapter" data-level="7.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo-plotphylo"><i class="fa fa-check"></i><b>7.7</b> runPhylo() &amp; plotPhylo()</a></li>
<li class="chapter" data-level="7.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>7.8</b> plotFreq()</a></li>
<li class="chapter" data-level="7.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>7.9</b> calcConsensus()</a></li>
<li class="chapter" data-level="7.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>7.10</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="7.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>7.11</b> plotGeneCopy()</a></li>
<li class="chapter" data-level="7.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotalluvial"><i class="fa fa-check"></i><b>7.12</b> plotAlluvial()</a></li>
<li class="chapter" data-level="7.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runpca"><i class="fa fa-check"></i><b>7.13</b> runPca()</a>
<ul>
<li class="chapter" data-level="7.13.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotscree"><i class="fa fa-check"></i><b>7.13.1</b> plotScree</a></li>
<li class="chapter" data-level="7.13.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotpca"><i class="fa fa-check"></i><b>7.13.2</b> plotPCA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="merging-copykit-objects.html"><a href="merging-copykit-objects.html"><i class="fa fa-check"></i><b>8</b> Merging CopyKit Objects</a></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a></li>
<li class="chapter" data-level="10" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>10</b> FAQ</a>
<ul>
<li class="chapter" data-level="10.1" data-path="faq.html"><a href="faq.html#error-unable-to-load-package-after-installation."><i class="fa fa-check"></i><b>10.1</b> Error unable to load package after installation.</a></li>
<li class="chapter" data-level="10.2" data-path="faq.html"><a href="faq.html#copykit-is-very-slow-despite-parallel-processing"><i class="fa fa-check"></i><b>10.2</b> CopyKit is very slow despite parallel processing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="learn-more.html"><a href="learn-more.html"><i class="fa fa-check"></i><b>11</b> Learn more</a>
<ul>
<li class="chapter" data-level="11.1" data-path="learn-more.html"><a href="learn-more.html#copy-number-data-from-short-reads-sequencing"><i class="fa fa-check"></i><b>11.1</b> Copy Number data from short reads sequencing</a></li>
<li class="chapter" data-level="11.2" data-path="learn-more.html"><a href="learn-more.html#principle-lm"><i class="fa fa-check"></i><b>11.2</b> Principle</a></li>
<li class="chapter" data-level="11.3" data-path="learn-more.html"><a href="learn-more.html#varbin-lm"><i class="fa fa-check"></i><b>11.3</b> VarBin</a></li>
<li class="chapter" data-level="11.4" data-path="learn-more.html"><a href="learn-more.html#gccor-lm"><i class="fa fa-check"></i><b>11.4</b> GC correction</a></li>
<li class="chapter" data-level="11.5" data-path="learn-more.html"><a href="learn-more.html#mergelevels-lm"><i class="fa fa-check"></i><b>11.5</b> Merge Levels</a></li>
<li class="chapter" data-level="11.6" data-path="learn-more.html"><a href="learn-more.html#infercn-lm"><i class="fa fa-check"></i><b>11.6</b> How do we infer copy numbers</a></li>
<li class="chapter" data-level="11.7" data-path="learn-more.html"><a href="learn-more.html#segmentation-lm"><i class="fa fa-check"></i><b>11.7</b> Segmentation</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="learn-more.html"><a href="learn-more.html#cbs-lm"><i class="fa fa-check"></i><b>11.7.1</b> CBS</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pre-processing" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Pre-processing<a href="pre-processing.html#pre-processing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>CopyKit’s pre-processing module workhorse is <code>runVarbin()</code></p>
<p><code>runVarbin()</code> is a wrapper for a series of functions that perform three main processes:</p>
<ol style="list-style-type: decimal">
<li>Binning and read counting from the <code>.bam</code> files</li>
<li>Applying a variance stabilization transformation to the bin counts</li>
<li>Piece-wise segmentating stabilized bin counts</li>
</ol>
<p>We load CopyKit with the <code>library()</code> function.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="pre-processing.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(copykit)</span></code></pre></div>
<p><strong>NOTE:</strong> If the BiocParallel framework was registered, CopyKit functions will, whenever possible, run in parallel.
For more information check the <a href="parallelization.html#parallelization">parallelization</a> section.</p>
<div id="from-single-cell-bam-files" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> From Single Cell BAM files<a href="pre-processing.html#from-single-cell-bam-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="runvarbin" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> runVarbin()<a href="pre-processing.html#runvarbin" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The runVarbin() function takes in the file path of the folder where your <em>marked duplicate BAM files are located</em>.</p>
<p>Each cell must be its own BAM file. For Chromium single cell CNA users (10X Genomics), this means splitting the possorted.bam file by cell barcode into individual BAM files.</p>
<p>If files are originated from <em>paired-end</em> sequencing, please ensure that the argument “is_paired_end” is set to TRUE during `runVarbin().</p>
<p>Once run, <code>runVarbin()</code> uses the variable binning method to count the number of reads in each genomic bin. - <a href="learn-more.html#varbin-lm">Learn More!</a>.</p>
<p>A <a href="https://snakemake.readthedocs.io/en/stable/">snakemake</a> pipeline is provided <a href="https://github.com/navinlabcode/copykit/tree/master/snakemake_pipelines">here</a> to convert FASTQ files to marked BAM files, but it is not a required step. If you already have marked BAM files from another source, you can skip this step. Note that the reads must be aligned to the same genome assembly as used in CopyKit.</p>
<p>The input for <code>runVarbin()</code> is the path of the folder containing your <em>marked duplicate <code>.bam</code> files</em>. An optional second argument <code>remove_Y</code> provides a convenient shortcut to exclude chromosome Y from the dataset.</p>
<pre><code>To learn about all arguments that runVarbin() accepts use ?runVarbin</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="pre-processing.html#cb9-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;~/path/to/bam/files/&quot;</span>, </span>
<span id="cb9-2"><a href="pre-processing.html#cb9-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">remove_Y =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Counting reads for genome hg38 and resolution: 220kb</code></pre>
<pre><code>## 34 bam files had less than 10 mean bincounts and were removed.</code></pre>
<pre><code>## Performing GC correction.</code></pre>
<pre><code>## Running variance stabilization transformation: ft</code></pre>
<pre><code>## Smoothing outlier bins.</code></pre>
<pre><code>## Running segmentation algorithm: CBS for genome hg38</code></pre>
<pre><code>## Merging levels.</code></pre>
<pre><code>## Done.</code></pre>
<p>The <code>runVarbin()</code> function uses the hg38 genome assembly and a 220kb bin resolution by default, but these can be adjusted using the <code>"genome"</code> or <code>"resolution"</code> arguments. If your BAM files come from paired-end sequencing, be sure to set the <code>"is_paired_end"</code> argument to TRUE. (See the function documentation for more information: ?runVarbin)</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="pre-processing.html#cb18-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runVarbin</span>(<span class="st">&quot;~/path/to/bam/files/&quot;</span>, </span>
<span id="cb18-2"><a href="pre-processing.html#cb18-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">remove_Y =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-3"><a href="pre-processing.html#cb18-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">is_paired_end =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>The resulting object is the CopyKit object.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="pre-processing.html#cb19-1" aria-hidden="true" tabindex="-1"></a>tumor</span></code></pre></div>
<pre><code>## class: CopyKit 
## dim: 11268 1502 
## metadata(3): genome resolution vst
## assays(6): bincounts ft ... ratios logr
## rownames(11268): 1 2 ... 11267 11268
## rowData names(3): gc_content abspos arm
## colnames(1502): PMTC6LiverC100DL1S2_S100_L001_R1_001
##   PMTC6LiverC100DL1S6_S484_L002_R1_001 ... PMTC6LiverC9DL1S5_S393_L002_R1_001
##   PMTC6LiverC9DL6L7S1_S1161_L004_R1_001
## colData names(9): sample reads_assigned_bins ... reads_total
##   percentage_duplicates
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowRanges has: 11268 ranges
## Phylo: Phylogenetic tree with 0 tips and  nodes
## consensus dim: 0 0</code></pre>
<p>CopyKit objects inherit from the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/SingleCellExperiment/inst/doc/intro.html">SingleCellExperiment</a> class.</p>
<p>Each column corresponds to a single cell and each row corresponds to a bin. The bincounts assay, which can be accessed using the <code>bincounts()</code> function, stores the read counts for each bin.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="pre-processing.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">bincounts</span>(tumor)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## # A tibble: 6 × 5
##   PMTC6LiverC100DL1S2_S… PMTC6LiverC100D… PMTC6LiverC100D… PMTC6LiverC100D… PMTC6LiverC101D…
##                    &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;            &lt;dbl&gt;
## 1                   37.1            10.3              44.5             44.2             38.6
## 2                   28.8            12.8              53.1             52.0             31.4
## 3                   34.8             9.03             50.4             47.3             31.9
## 4                   24.3            14.2              34.9             35.5             36.9
## 5                   30.8             9.18             55.8             37.1             29.2
## 6                   29.3             5.88             43.0             43.1             38.3</code></pre>
<p>To address overdispersion, CopyKit performs a variance stabilization transformation (VST) of the count matrix using the Freeman-Tukey transformation. Alternatively, log transformation is also available (see the function documentation for more information ?runVst())</p>
<p>The resulting transformation is stored within the <em>ft</em> assay, which can be accessed with <code>assay()</code>:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="pre-processing.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(tumor, <span class="st">&#39;ft&#39;</span>)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>##   PMTC6LiverC100DL1S2_S100_L001_R1_001 PMTC6LiverC100DL1S6_S484_L002_R1_001
## 1                            12.258603                             6.558716
## 2                            10.832889                             7.300770
## 3                            11.880779                             6.172013
## 4                             9.951377                             7.674830
## 5                            11.181771                             6.220463
## 6                            10.921155                             5.047847
##   PMTC6LiverC100DL4L5S1_S868_L003_R1_001 PMTC6LiverC100DL6L7S3_S1252_L004_R1_001
## 1                               13.42067                                13.36691
## 2                               14.63815                                14.48431
## 3                               14.27288                                13.82443
## 4                               11.89424                                12.00304
## 5                               15.01185                                12.26513
## 6                               13.19827                                13.20130
##   PMTC6LiverC101DL1S2_S101_L001_R1_001
## 1                             12.50574
## 2                             11.30275
## 3                             11.37859
## 4                             12.23737
## 5                             10.90282
## 6                             12.46409</code></pre>
<p>We use segmentation to divide the genome-ordered bin counts into piecewise constant segments, and the means of these segments can be used to infer copy number states across the genomic regions.</p>
<p>The runVarbin() function allows for the selection of different segmentation methods using the <code>method</code> argument, which can be set to either <code>CBS</code> or <code>multipcf</code>. By default, the function uses the Circular Binary Segmentation (CBS) - <a href="learn-more.html#cbs-lm">Learn More!</a> - method from the <a href="https://bioconductor.org/packages/release/bioc/html/DNAcopy.html">DNAcopy</a> package. An additional argument, <code>alpha</code>, controls the significance levels required to accept change-points.</p>
<p>The second segmentation option is the Multi-sample Piecewise Constant Fit (multipcf) segmentation from the <a href="https://www.bioconductor.org/packages/release/bioc/html/copynumber.html">copynumber</a> package, which performs a joint segmentation of the samples and identifies common breakpoints across all samples.</p>
<p>The resulting segmentation is stored within the CopyKit object into two different assays: <em>ratios</em> and <em>segment_ratios</em>. <a href="learn-more.html#infercn-lm">Learn More!</a> which can be accessed with the functions <code>ratios()</code> and <code>segment_ratios()</code></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="pre-processing.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">ratios</span>(tumor)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>##   PMTC6LiverC100DL1S2_S100_L001_R1_001 PMTC6LiverC100DL1S6_S484_L002_R1_001
## 1                                 1.15                                 0.90
## 2                                 0.89                                 1.13
## 3                                 1.08                                 0.80
## 4                                 0.75                                 1.25
## 5                                 0.95                                 0.81
## 6                                 0.91                                 0.52
##   PMTC6LiverC100DL4L5S1_S868_L003_R1_001 PMTC6LiverC100DL6L7S3_S1252_L004_R1_001
## 1                                   0.94                                    1.06
## 2                                   1.12                                    1.25
## 3                                   1.06                                    1.14
## 4                                   0.74                                    0.86
## 5                                   1.18                                    0.89
## 6                                   0.91                                    1.04
##   PMTC6LiverC101DL1S2_S101_L001_R1_001
## 1                                 1.03
## 2                                 0.84
## 3                                 0.85
## 4                                 0.99
## 5                                 0.78
## 6                                 1.03</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="pre-processing.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">segment_ratios</span>(tumor)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>##   PMTC6LiverC100DL1S2_S100_L001_R1_001 PMTC6LiverC100DL1S6_S484_L002_R1_001
## 1                                    1                                 0.99
## 2                                    1                                 0.99
## 3                                    1                                 0.99
## 4                                    1                                 0.99
## 5                                    1                                 0.99
## 6                                    1                                 0.99
##   PMTC6LiverC100DL4L5S1_S868_L003_R1_001 PMTC6LiverC100DL6L7S3_S1252_L004_R1_001
## 1                                      1                                    1.03
## 2                                      1                                    1.03
## 3                                      1                                    1.03
## 4                                      1                                    1.03
## 5                                      1                                    1.03
## 6                                      1                                    1.03
##   PMTC6LiverC101DL1S2_S101_L001_R1_001
## 1                                    1
## 2                                    1
## 3                                    1
## 4                                    1
## 5                                    1
## 6                                    1</code></pre>
<hr />
</div>
</div>
<div id="runvarbin-modules" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> runVarbin() modules<a href="pre-processing.html#runvarbin-modules" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>runVarbin()</code> module includes the following functions, which are detailed below.
<strong>It is important to note that if you have already started from runVarbin(), you do not need to run these functions again.</strong>
The details of these functions is to enable you to run the modules under different conditions without requiring you to re-run all <code>runVarbin()</code> module.</p>
<div id="runcountreads" class="section level4 hasAnchor" number="4.2.0.1">
<h4><span class="header-section-number">4.2.0.1</span> runCountReads()<a href="pre-processing.html#runcountreads" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>runCountReads()</code> function counts the reads into bins, and performs GC correction of the counts - <a href="learn-more.html#gccor-lm">Learn More!</a>.</p>
<p>The genome argument defines the genome assembly (“hg38”, “hg19”) and the resolution argument specifies the size of the variable bins (“55kb”, “110kb”, “195kb”, “220kb”, “280kb”, “500kb”, “1Mb”, “2.8Mb”).
The <code>remove_Y</code> argument provides a convenient shortcut to exclude the chromosome Y from the dataset.</p>
</div>
<div id="runvst" class="section level4 hasAnchor" number="4.2.0.2">
<h4><span class="header-section-number">4.2.0.2</span> runVst()<a href="pre-processing.html#runvst" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>runVst()</code> function performs variance stabilization transformation on the bin counts, using either the freeman-tukey (‘ft’) or ‘log’ transformation.</p>
</div>
<div id="runsegmentation" class="section level4 hasAnchor" number="4.2.0.3">
<h4><span class="header-section-number">4.2.0.3</span> runSegmentation()<a href="pre-processing.html#runsegmentation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>runSegmentation()</code> function runs either the ‘CBS’ or ‘multipcf’ segmentation, and then merges consecutive segments that don’t reach the significance threshold <a href="learn-more.html#mergelevels-lm">Learn More!</a>.
The argument <em>alpha</em> controls the significance threshold for <code>CBS</code> segmentation, whil th <em>gamma</em> argument controls the penalty for the <code>multipcf</code> segmentation.</p>
</div>
<div id="lognorm" class="section level4 hasAnchor" number="4.2.0.4">
<h4><span class="header-section-number">4.2.0.4</span> logNorm()<a href="pre-processing.html#lognorm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><code>logNorm()</code> performs a log transformation of the <em>segment_ratios</em> assay and stores into the <em>logr</em> assay. The <em>logr</em> assay is used downstream by functions such as <code>runUmap()</code>.</p>
<hr />
</div>
</div>
<div id="from-external-count-data-or-user-defined-scaffolds" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> From external count data or user-defined scaffolds<a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To use CopyKit’s downstream functions with processed datasets, we can create a <code>CopyKit</code> object meeting the following requirements:</p>
<ol style="list-style-type: decimal">
<li><p>Either a bin count matrix where columns represent cells and rows represent bins (to be stored in the <code>bincounts</code> assay) or a segment mean ratios matrix where columns represent cells and rows represent segment mean values for each bin. An integer matrix of copy number calls can also be used at this step (to be stored in the <code>segment_ratios</code> assay).</p></li>
<li><p>A <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html">GenomicRanges</a> object with length equal to the number of bins from the cell assay matrix. As long as the length requirement is respected this allows user-defined scaffolds to be used within CopyKit.</p></li>
</ol>
<p>To construct the CopyKit object, the following mock code is presented as an example.
If providing bincounts:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="pre-processing.html#cb29-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">CopyKit</span>(<span class="fu">list</span>(<span class="at">bincounts =</span> cell_bincount_matrix),</span>
<span id="cb29-2"><a href="pre-processing.html#cb29-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">rowRanges =</span> genomic_ranges_scaffold)</span></code></pre></div>
<p>If providing bincounts, the functions <code>runVst()</code>, <code>runSegmentation()</code>, and <code>logNorm()</code> can be used to continue with the analysis. To obtain variance stabilized counts, and segment ratio means.</p>
<p>If providing segment mean ratios:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="pre-processing.html#cb30-1" aria-hidden="true" tabindex="-1"></a>obj <span class="ot">&lt;-</span> <span class="fu">CopyKit</span>(<span class="fu">list</span>(<span class="at">segment_ratios =</span> cell_bincount_matrix),</span>
<span id="cb30-2"><a href="pre-processing.html#cb30-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">rowRanges =</span> genomic_ranges_scaffold)</span></code></pre></div>
<p>It is useful to call <code>logNorm()</code> and set the <em>logr</em> slot for downstream functions.
The resulting object can then be passed on to the Quality Control and Analysis modules of CopyKit.</p>
<p>Furthermore, it is useful for downstream functions to add the genome assembly used on the dataset.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="pre-processing.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(obj)<span class="sc">$</span>genome <span class="ot">&lt;-</span> <span class="st">&quot;hg38&quot;</span></span></code></pre></div>
<p><strong>NOTE:</strong> If no bincount matrix is provided, functions that require a matrix of bincounts, such as <code>runMetrics()</code> can’t be used with this object.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quick-start.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="qc-module.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
