<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 QC module | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="5 QC module | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 QC module | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pre-processing.html"/>
<link rel="next" href="analysis-and-visualization-module.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#from-bam-files"><i class="fa fa-check"></i><b>4.1</b> From BAM files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1.1</b> runVarbin()</a></li>
<li class="chapter" data-level="4.1.2" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin-modules"><i class="fa fa-check"></i><b>4.1.2</b> runVarbin() modules</a></li>
<li class="chapter" data-level="4.1.3" data-path="pre-processing.html"><a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds"><i class="fa fa-check"></i><b>4.1.3</b> From external count data or user-defined scaffolds</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findaneuploidcells"><i class="fa fa-check"></i><b>5.2</b> findAneuploidCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#findoutliers"><i class="fa fa-check"></i><b>5.3</b> findOutliers()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>6</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>6.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>6.2</b> plotRatio()</a></li>
<li class="chapter" data-level="6.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>6.3</b> runUmap()</a></li>
<li class="chapter" data-level="6.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>6.4</b> plotUmap()</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>6.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>6.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="6.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>6.5.2</b> findClusters()</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo"><i class="fa fa-check"></i><b>6.6</b> runPhylo()</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotphylo"><i class="fa fa-check"></i><b>6.7</b> plotPhylo()</a></li>
<li class="chapter" data-level="6.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcinteger"><i class="fa fa-check"></i><b>6.8</b> calcInteger()</a></li>
<li class="chapter" data-level="6.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>6.9</b> plotHeatmap()</a></li>
<li class="chapter" data-level="6.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>6.10</b> plotFreq()</a></li>
<li class="chapter" data-level="6.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>6.11</b> calcConsensus()</a></li>
<li class="chapter" data-level="6.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>6.12</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="6.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>6.13</b> plotGeneCopy()</a></li>
<li class="chapter" data-level="6.14" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotalluvial"><i class="fa fa-check"></i><b>6.14</b> plotAlluvial()</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learn-more.html"><a href="learn-more.html"><i class="fa fa-check"></i><b>7</b> Learn more</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learn-more.html"><a href="learn-more.html#copy-number-data-from-short-reads-sequencing"><i class="fa fa-check"></i><b>7.1</b> Copy Number data from short reads sequencing</a></li>
<li class="chapter" data-level="7.2" data-path="learn-more.html"><a href="learn-more.html#principle-lm"><i class="fa fa-check"></i><b>7.2</b> Principle</a></li>
<li class="chapter" data-level="7.3" data-path="learn-more.html"><a href="learn-more.html#varbin-lm"><i class="fa fa-check"></i><b>7.3</b> VarBin</a></li>
<li class="chapter" data-level="7.4" data-path="learn-more.html"><a href="learn-more.html#gccor-lm"><i class="fa fa-check"></i><b>7.4</b> GC correction</a></li>
<li class="chapter" data-level="7.5" data-path="learn-more.html"><a href="learn-more.html#mergelevels-lm"><i class="fa fa-check"></i><b>7.5</b> Merge Levels</a></li>
<li class="chapter" data-level="7.6" data-path="learn-more.html"><a href="learn-more.html#segmentation-lm"><i class="fa fa-check"></i><b>7.6</b> Segmentation</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="learn-more.html"><a href="learn-more.html#cbs-lm"><i class="fa fa-check"></i><b>7.6.1</b> CBS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>8</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qc-module" class="section level1" number="5">
<h1><span class="header-section-number">5</span> QC module</h1>
<p>CopyKit Quality Control Module consists of 3 main functions:</p>
<ol style="list-style-type: decimal">
<li><p><code>runMetrics()</code></p></li>
<li><p><code>findOutliers()</code></p></li>
<li><p><code>findAneuploidCells()</code>.</p></li>
</ol>
<div id="runmetrics" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> runMetrics()</h2>
<p><code>runMetrics()</code> adds basic quality control information to colData.
It returns sample-wise metrics of overdispersion and breakpoint counts.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="qc-module.html#cb54-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">runMetrics</span>(tumor)</span></code></pre></div>
<pre><code>## Calculating overdispersion.</code></pre>
<pre><code>## Counting breakpoints.</code></pre>
<pre><code>## Done.</code></pre>
<p>The resulting information can be viewed with:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="qc-module.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tumor)</span></code></pre></div>
<pre><code>## DataFrame with 1502 rows and 11 columns
##                                                         sample reads_assigned_bins reads_unmapped
##                                                    &lt;character&gt;           &lt;integer&gt;      &lt;integer&gt;
## PMTC6LiverC100DL1S2_S100_L001_R1_001    PMTC6LiverC100DL1S2_..              362665          30250
## PMTC6LiverC100DL1S6_S484_L002_R1_001    PMTC6LiverC100DL1S6_..              130570          22260
## PMTC6LiverC100DL4L5S1_S868_L003_R1_001  PMTC6LiverC100DL4L5S..              536017          30942
## PMTC6LiverC100DL6L7S3_S1252_L004_R1_001 PMTC6LiverC100DL6L7S..              467002          28239
## PMTC6LiverC101DL1S2_S101_L001_R1_001    PMTC6LiverC101DL1S2_..              423654          28756
## ...                                                        ...                 ...            ...
## PMTC6LiverC99DL4L5S1_S867_L003_R1_001   PMTC6LiverC99DL4L5S1..              460498          37945
## PMTC6LiverC99DL6L7S3_S1251_L004_R1_001  PMTC6LiverC99DL6L7S3..              306486          28090
## PMTC6LiverC9DL1S1_S9_L001_R1_001        PMTC6LiverC9DL1S1_S9..              274402          38326
## PMTC6LiverC9DL1S5_S393_L002_R1_001      PMTC6LiverC9DL1S5_S3..              465001          34111
## PMTC6LiverC9DL6L7S1_S1161_L004_R1_001   PMTC6LiverC9DL6L7S1_..              336559          38814
##                                         reads_duplicates reads_multimapped reads_unassigned reads_ambiguous
##                                                &lt;integer&gt;         &lt;integer&gt;        &lt;integer&gt;       &lt;integer&gt;
## PMTC6LiverC100DL1S2_S100_L001_R1_001               34883                 0            75555              90
## PMTC6LiverC100DL1S6_S484_L002_R1_001               12657                 0            28651              27
## PMTC6LiverC100DL4L5S1_S868_L003_R1_001             58222                 0           110352             103
## PMTC6LiverC100DL6L7S3_S1252_L004_R1_001            53055                 0            98560              91
## PMTC6LiverC101DL1S2_S101_L001_R1_001               43008                 0            87786              76
## ...                                                  ...               ...              ...             ...
## PMTC6LiverC99DL4L5S1_S867_L003_R1_001              54159                 0            96735              86
## PMTC6LiverC99DL6L7S3_S1251_L004_R1_001             36534                 0            64866              56
## PMTC6LiverC9DL1S1_S9_L001_R1_001                   25527                 0            59107              54
## PMTC6LiverC9DL1S5_S393_L002_R1_001                 48055                 0            96126             109
## PMTC6LiverC9DL6L7S1_S1161_L004_R1_001              37898                 0            71998              60
##                                         reads_total percentage_duplicates overdispersion breakpoint_count
##                                           &lt;numeric&gt;             &lt;numeric&gt;      &lt;numeric&gt;        &lt;numeric&gt;
## PMTC6LiverC100DL1S2_S100_L001_R1_001         503443                 0.069     0.00262992                0
## PMTC6LiverC100DL1S6_S484_L002_R1_001         194165                 0.065     0.01564945                0
## PMTC6LiverC100DL4L5S1_S868_L003_R1_001       735636                 0.079     0.00302844                0
## PMTC6LiverC100DL6L7S3_S1252_L004_R1_001      646947                 0.082     0.00379236                0
## PMTC6LiverC101DL1S2_S101_L001_R1_001         583280                 0.074     0.00674672                0
## ...                                             ...                   ...            ...              ...
## PMTC6LiverC99DL4L5S1_S867_L003_R1_001        649423                 0.083     0.00195059                0
## PMTC6LiverC99DL6L7S3_S1251_L004_R1_001       436032                 0.084     0.00362179                0
## PMTC6LiverC9DL1S1_S9_L001_R1_001             397416                 0.064     0.00406539                0
## PMTC6LiverC9DL1S5_S393_L002_R1_001           643402                 0.075     0.00276767                0
## PMTC6LiverC9DL6L7S1_S1161_L004_R1_001        485329                 0.078     0.00231466                0</code></pre>
</div>
<div id="findaneuploidcells" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> findAneuploidCells()</h2>
<p>Datasets may contain euploid cells mixed with the aneuploidy cells.</p>
<p>To detect euploid cells <code>findAneuploidCells()</code> calculates the sample-wise coefficient of variation from the segment ratio means.
The expected coefficient of variation for euploid cells <code>N(0, 0.01)</code> is simulated for <em>x</em> data points, where <em>x</em> is equal to the number of cells within the dataset.
An expectation-maximization algorithm is used to fit a mixture of gaussian distributions to the coefficient of variation from the samples together with the simulated datasets.
The distribution containing the simulated dataset is inferred to be the euploid distribution.
Samples that group with the inferred euploid distribution and present coefficient of variation smaller than 5 standard deviations from the mean euploid distribution are classified as euploid samples.</p>
<p>The threshold can be changed from the automatic detection to a custom threshold with the argument <code>resolution</code>. For example, by setting a threshold = 0.1, findAneuploidCells will mark as euploid all cells with a coefficient of variation less or equal than 0.1.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="qc-module.html#cb60-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findAneuploidCells</span>(tumor)</span></code></pre></div>
<pre><code>## number of iterations= 17</code></pre>
<pre><code>## Copykit detected 612 that are possibly normal cells using a resolution of: 0.054</code></pre>
<pre><code>## Added information to colData(CopyKit).</code></pre>
<p>The results from <code>findAneuploidCells()</code> are stored within the colData in the column <em>is_aneuploid</em>.</p>
<p>We can visualize the results with <code>plotHeatmap()</code>:</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="qc-module.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;is_aneuploid&#39;</span>, <span class="at">row_split =</span> <span class="st">&#39;is_aneuploid&#39;</span>, <span class="at">n_threads =</span> <span class="dv">40</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 40 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/find_normal_heat-1.png" width="672" /></p>
<p>The object is subsetted in the same way as with any R object, to keep only the aneuploid cells.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="qc-module.html#cb73-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> tumor[,<span class="fu">colData</span>(tumor)<span class="sc">$</span>is_aneuploid <span class="sc">==</span> <span class="cn">TRUE</span>]</span></code></pre></div>
</div>
<div id="findoutliers" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> findOutliers()</h2>
<p><code>findOutliers()</code> annotates low-quality cells according to a defined resolution threshold.</p>
<p>To detect low-quality samples, CopyKit calculates the Pearson correlation matrix of all samples from the segment ratio means.
Next, we calculate a sample-wise mean of the correlation between a cell and its k-nearest-neighbors (default = 5).
Cells in which the correlation value is lower than the defined threshold are classified as low-quality cells (default = 0.9).</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="qc-module.html#cb74-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> <span class="fu">findOutliers</span>(tumor)</span></code></pre></div>
<pre><code>## Calculating correlation matrix.</code></pre>
<pre><code>## Marked 101 cells as outliers.</code></pre>
<pre><code>## Adding information to metadata. Access with colData(scCNA).</code></pre>
<pre><code>## Done.</code></pre>
<p>The default correlation cutoff for filtering can be adjusted with the argument ‘resolution.’ For example, setting the resolution = 0.8 will mark all cells with a mean correlation smaller than 0.8 as low-quality cells.
Higher resolution values will result in stricter filtering criterias.</p>
<p>Results from <code>findOutliers()</code> are added to colData (column <em>outlier</em>) marking cells that can be removed or kept.</p>
<p>We can check the results with <code>plotHeatmap()</code>. To make visualization easier, rows can also be split according to elements of colData with the argument <code>row_split</code>.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="qc-module.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotHeatmap</span>(tumor, <span class="at">label =</span> <span class="st">&#39;outlier&#39;</span>, <span class="at">row_split =</span> <span class="st">&#39;outlier&#39;</span>, <span class="at">n_threads =</span> <span class="dv">40</span>)</span></code></pre></div>
<pre><code>## Ordering by consensus requires cluster information.</code></pre>
<pre><code>## Switching to hclust.</code></pre>
<pre><code>## No distance matrix detected in the scCNA object.</code></pre>
<pre><code>## Calculating distance matrix with metric: euclidean</code></pre>
<pre><code>## Using 40 cores.</code></pre>
<pre><code>## Access distance matrix with copykit::distMat()</code></pre>
<pre><code>## Done.</code></pre>
<pre><code>## Plotting Heatmap.</code></pre>
<p><img src="CopyKit-UserGuide_files/figure-html/filter_cells_heat-1.png" width="672" /></p>
<p>We remove the marked noise cells from the object with:</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="qc-module.html#cb88-1" aria-hidden="true" tabindex="-1"></a>tumor <span class="ot">&lt;-</span> tumor[,<span class="fu">colData</span>(tumor)<span class="sc">$</span>outlier <span class="sc">==</span> <span class="cn">FALSE</span>]</span></code></pre></div>
<p>The dataset should be ready to proceed with the analysis.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pre-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-and-visualization-module.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
