<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Session Info | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Session Info | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Session Info | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="learn-more.html"/>

<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#from-bam-files"><i class="fa fa-check"></i><b>4.1</b> From BAM files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1.1</b> runVarbin()</a></li>
<li class="chapter" data-level="4.1.2" data-path="pre-processing.html"><a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds"><i class="fa fa-check"></i><b>4.1.2</b> From external count data or user-defined scaffolds</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findnormalcells"><i class="fa fa-check"></i><b>5.2</b> findNormalCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#filtercells"><i class="fa fa-check"></i><b>5.3</b> filterCells()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>6</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>6.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>6.2</b> plotRatio()</a></li>
<li class="chapter" data-level="6.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>6.3</b> runUmap()</a></li>
<li class="chapter" data-level="6.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>6.4</b> plotUmap()</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>6.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>6.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="6.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>6.5.2</b> findClusters()</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo"><i class="fa fa-check"></i><b>6.6</b> runPhylo()</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotphylo"><i class="fa fa-check"></i><b>6.7</b> plotPhylo()</a></li>
<li class="chapter" data-level="6.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcinteger"><i class="fa fa-check"></i><b>6.8</b> calcInteger()</a></li>
<li class="chapter" data-level="6.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>6.9</b> plotHeatmap()</a></li>
<li class="chapter" data-level="6.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>6.10</b> plotFreq()</a></li>
<li class="chapter" data-level="6.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>6.11</b> calcConsensus()</a></li>
<li class="chapter" data-level="6.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>6.12</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="6.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findvariablegenes"><i class="fa fa-check"></i><b>6.13</b> findVariableGenes()</a></li>
<li class="chapter" data-level="6.14" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotvariablegenes"><i class="fa fa-check"></i><b>6.14</b> plotVariableGenes()</a></li>
<li class="chapter" data-level="6.15" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>6.15</b> plotGeneCopy()</a></li>
<li class="chapter" data-level="6.16" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotalluvial"><i class="fa fa-check"></i><b>6.16</b> plotAlluvial()</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="learn-more.html"><a href="learn-more.html"><i class="fa fa-check"></i><b>7</b> Learn more</a>
<ul>
<li class="chapter" data-level="7.1" data-path="learn-more.html"><a href="learn-more.html#copy-number-data-from-short-reads-sequencing"><i class="fa fa-check"></i><b>7.1</b> Copy Number data from short reads sequencing</a></li>
<li class="chapter" data-level="7.2" data-path="learn-more.html"><a href="learn-more.html#principle-lm"><i class="fa fa-check"></i><b>7.2</b> Principle</a></li>
<li class="chapter" data-level="7.3" data-path="learn-more.html"><a href="learn-more.html#varbin-lm"><i class="fa fa-check"></i><b>7.3</b> VarBin</a></li>
<li class="chapter" data-level="7.4" data-path="learn-more.html"><a href="learn-more.html#gccor-lm"><i class="fa fa-check"></i><b>7.4</b> GC correction</a></li>
<li class="chapter" data-level="7.5" data-path="learn-more.html"><a href="learn-more.html#mergelevels-lm"><i class="fa fa-check"></i><b>7.5</b> Merge Levels</a></li>
<li class="chapter" data-level="7.6" data-path="learn-more.html"><a href="learn-more.html#segmentation-lm"><i class="fa fa-check"></i><b>7.6</b> Segmentation</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="learn-more.html"><a href="learn-more.html#cbs-lm"><i class="fa fa-check"></i><b>7.6.1</b> CBS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>8</b> Session Info</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Session Info</h1>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="session-info.html#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] copykit_0.0.0.9031          DNAcopy_1.68.0              Rsubread_2.8.1             
##  [4] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0 Biobase_2.54.0             
##  [7] GenomicRanges_1.46.0        GenomeInfoDb_1.30.0         IRanges_2.28.0             
## [10] S4Vectors_0.32.2            BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
## [13] matrixStats_0.61.0          BiocParallel_1.28.0         bookdown_0.24              
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3            rtracklayer_1.54.0        prabclus_2.3-2           
##   [4] coda_0.19-4               tidyr_1.1.4               ggplot2_3.3.5            
##   [7] clusterGeneration_1.3.7   bit64_4.0.5               knitr_1.36               
##  [10] irlba_2.3.3               DelayedArray_0.20.0       KEGGREST_1.34.0          
##  [13] RCurl_1.98-1.5            doParallel_1.0.16         generics_0.1.1           
##  [16] GenomicFeatures_1.46.1    ScaledMatrix_1.2.0        callr_3.7.0              
##  [19] cowplot_1.1.1             usethis_2.1.3             RSQLite_2.2.8            
##  [22] combinat_0.0-8            bit_4.0.4                 xml2_1.3.2               
##  [25] lubridate_1.8.0           httpuv_1.6.3              assertthat_0.2.1         
##  [28] viridis_0.6.2             amap_0.8-18               xfun_0.28                
##  [31] hms_1.1.1                 jquerylib_0.1.4           evaluate_0.14            
##  [34] promises_1.2.0.1          DEoptimR_1.0-9            fansi_0.5.0              
##  [37] restfulr_0.0.13           progress_1.2.2            dbplyr_2.1.1             
##  [40] igraph_1.2.8              DBI_1.1.1                 tmvnsim_1.0-2            
##  [43] paletteer_1.4.0           purrr_0.3.4               ellipsis_0.3.2           
##  [46] RSpectra_0.16-0           ggnewscale_0.4.5          dplyr_1.0.7              
##  [49] backports_1.3.0           biomaRt_2.50.0            sparseMatrixStats_1.6.0  
##  [52] vctrs_0.3.8               ggalluvial_0.12.3         remotes_2.4.1            
##  [55] cachem_1.0.6              withr_2.4.2               robustbase_0.93-9        
##  [58] GenomicAlignments_1.30.0  treeio_1.18.0             prettyunits_1.1.1        
##  [61] scran_1.22.0              mclust_5.4.7              mnormt_2.0.2             
##  [64] phytools_0.7-90           cluster_2.1.2             segmented_1.3-4          
##  [67] ape_5.5                   lazyeval_0.2.2            crayon_1.4.2             
##  [70] labeling_0.4.2            edgeR_3.36.0              pkgconfig_2.0.3          
##  [73] vipor_0.4.5               nlme_3.1-153              pkgload_1.2.3            
##  [76] nnet_7.3-16               pals_1.7                  devtools_2.4.2           
##  [79] diptest_0.76-0            rlang_0.4.12              miniUI_0.1.1.1           
##  [82] lifecycle_1.0.1           filelock_1.0.2            dbscan_1.1-8             
##  [85] BiocFileCache_2.2.0       rsvd_1.0.5                dichromat_2.0-0          
##  [88] rprojroot_2.0.2           Matrix_1.3-4              aplot_0.1.1              
##  [91] phangorn_2.7.1            boot_1.3-28               beeswarm_0.4.0           
##  [94] GlobalOptions_0.1.2       processx_3.5.2            viridisLite_0.4.0        
##  [97] png_0.1-7                 rjson_0.2.20              bitops_1.0-7             
## [100] Biostrings_2.62.0         blob_1.2.2                DelayedMatrixStats_1.16.0
## [103] shape_1.4.6               stringr_1.4.0             gridGraphics_0.5-1       
## [106] beachmat_2.10.0           scales_1.1.1              memoise_2.0.0            
## [109] magrittr_2.0.1            zlibbioc_1.40.0           compiler_4.1.1           
## [112] dqrng_0.3.0               BiocIO_1.4.0              RColorBrewer_1.1-2       
## [115] plotrix_3.8-2             clue_0.3-60               Rsamtools_2.10.0         
## [118] snakecase_0.11.0          cli_3.1.0                 XVector_0.34.0           
## [121] patchwork_1.1.1           ps_1.6.0                  mgcv_1.8-38              
## [124] MASS_7.3-54               tidyselect_1.1.1          stringi_1.7.5            
## [127] forcats_0.5.1             highr_0.9                 purler_0.1.1             
## [130] yaml_2.2.1                BiocSingular_1.10.0       locfit_1.5-9.4           
## [133] grid_4.1.1                sass_0.4.0                fastmatch_1.1-3          
## [136] tools_4.1.1               parallel_4.1.1            circlize_0.4.13          
## [139] rstudioapi_0.13           bluster_1.4.0             foreach_1.5.1            
## [142] gridExtra_2.3             janitor_2.1.0             metapod_1.2.0            
## [145] farver_2.1.0              scatterplot3d_0.3-41      digest_0.6.28            
## [148] FNN_1.1.3                 shiny_1.7.1               fpc_2.2-9                
## [151] quadprog_1.5-8            Rcpp_1.0.7                broom_0.7.10             
## [154] scuttle_1.4.0             later_1.3.0               httr_1.4.2               
## [157] AnnotationDbi_1.56.1      ComplexHeatmap_2.10.0     kernlab_0.9-29           
## [160] colorspace_2.0-2          XML_3.99-0.8              fs_1.5.0                 
## [163] reticulate_1.22           splines_4.1.1             uwot_0.1.10              
## [166] yulab.utils_0.0.4         statmod_1.4.36            rematch2_2.1.2           
## [169] tidytree_0.3.5            expm_0.999-6              aCGH_1.72.0              
## [172] multtest_2.50.0           flexmix_2.3-17            mapproj_1.2.7            
## [175] ggplotify_0.1.0           sessioninfo_1.2.1         xtable_1.8-4             
## [178] jsonlite_1.7.2            ggtree_3.2.0              modeltools_0.2-23        
## [181] ggfun_0.0.4               testthat_3.1.0            R6_2.5.1                 
## [184] pillar_1.6.4              htmltools_0.5.2           mime_0.12                
## [187] glue_1.4.2                fastmap_1.1.0             BiocNeighbors_1.12.0     
## [190] class_7.3-19              codetools_0.2-18          maps_3.4.0               
## [193] pkgbuild_1.2.0            utf8_1.2.2                lattice_0.20-45          
## [196] bslib_0.3.1               tibble_3.1.5              mixtools_1.2.0           
## [199] numDeriv_2016.8-1.1       ggbeeswarm_0.6.0          curl_4.3.2               
## [202] leiden_0.3.9              gtools_3.9.2              magick_2.7.3             
## [205] survival_3.2-13           limma_3.50.0              rmarkdown_2.11           
## [208] desc_1.4.0                munsell_0.5.0             GetoptLong_1.0.5         
## [211] fastcluster_1.2.3         GenomeInfoDbData_1.2.7    iterators_1.0.13         
## [214] gtable_0.3.0</code></pre>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Baslan, T., J. Kendall, L. Rodgers, H. Cox, M. Riggs, A. Stepansky, J. Troge, et al. 2012. <span>“Genome-Wide Copy Number Analysis of Single Cells.”</span> Journal Article. <em>Nat Protoc</em> 7 (6): 1024–41. <a href="https://doi.org/10.1038/nprot.2012.039">https://doi.org/10.1038/nprot.2012.039</a>.
</div>
<div class="csl-entry">
Gao, R., A. Davis, T. O. McDonald, E. Sei, X. Shi, Y. Wang, P. C. Tsai, et al. 2016. <span>“Punctuated Copy Number Evolution and Clonal Stasis in Triple-Negative Breast Cancer.”</span> Journal Article. <em>Nat Genet</em> 48 (10): 1119–30. <a href="https://doi.org/10.1038/ng.3641">https://doi.org/10.1038/ng.3641</a>.
</div>
<div class="csl-entry">
Navin, N., J. Kendall, J. Troge, P. Andrews, L. Rodgers, J. McIndoo, K. Cook, et al. 2011. <span>“Tumour Evolution Inferred by Single-Cell Sequencing.”</span> Journal Article. <em>Nature</em> 472 (7341): 90–94. <a href="https://doi.org/10.1038/nature09807">https://doi.org/10.1038/nature09807</a>.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="learn-more.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
