<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Session Info | CopyKit User Guide</title>
  <meta name="description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Session Info | CopyKit User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Session Info | CopyKit User Guide" />
  
  <meta name="twitter:description" content="This is a user guide for the package CopyKit: A toolkit for single-cell CNA analysis." />
  

<meta name="author" content="Darlan Conterno Minussi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="merging-copykit-objects.html"/>
<link rel="next" href="faq.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/copykit_hexsticker_v2.png"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="parallelization.html"><a href="parallelization.html"><i class="fa fa-check"></i><b>2</b> Parallelization</a></li>
<li class="chapter" data-level="3" data-path="quick-start.html"><a href="quick-start.html"><i class="fa fa-check"></i><b>3</b> Quick-Start</a></li>
<li class="chapter" data-level="4" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>4</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="pre-processing.html"><a href="pre-processing.html#from-single-cell-bam-files"><i class="fa fa-check"></i><b>4.1</b> From Single Cell BAM files</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin"><i class="fa fa-check"></i><b>4.1.1</b> runVarbin()</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pre-processing.html"><a href="pre-processing.html#runvarbin-modules"><i class="fa fa-check"></i><b>4.2</b> runVarbin() modules</a></li>
<li class="chapter" data-level="4.3" data-path="pre-processing.html"><a href="pre-processing.html#from-external-count-data-or-user-defined-scaffolds"><i class="fa fa-check"></i><b>4.3</b> From external count data or user-defined scaffolds</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qc-module.html"><a href="qc-module.html"><i class="fa fa-check"></i><b>5</b> QC module</a>
<ul>
<li class="chapter" data-level="5.1" data-path="qc-module.html"><a href="qc-module.html#runmetrics"><i class="fa fa-check"></i><b>5.1</b> runMetrics()</a></li>
<li class="chapter" data-level="5.2" data-path="qc-module.html"><a href="qc-module.html#findaneuploidcells"><i class="fa fa-check"></i><b>5.2</b> findAneuploidCells()</a></li>
<li class="chapter" data-level="5.3" data-path="qc-module.html"><a href="qc-module.html#findoutliers"><i class="fa fa-check"></i><b>5.3</b> findOutliers()</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="knn-smoothing.html"><a href="knn-smoothing.html"><i class="fa fa-check"></i><b>6</b> kNN smoothing</a></li>
<li class="chapter" data-level="7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html"><i class="fa fa-check"></i><b>7</b> Analysis and Visualization module</a>
<ul>
<li class="chapter" data-level="7.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotmetrics"><i class="fa fa-check"></i><b>7.1</b> plotMetrics()</a></li>
<li class="chapter" data-level="7.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotratio"><i class="fa fa-check"></i><b>7.2</b> plotRatio()</a></li>
<li class="chapter" data-level="7.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runumap"><i class="fa fa-check"></i><b>7.3</b> runUmap()</a></li>
<li class="chapter" data-level="7.4" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#scquantum-calcinteger"><i class="fa fa-check"></i><b>7.4</b> scquantum &amp; calcInteger()</a></li>
<li class="chapter" data-level="7.5" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#clustering"><i class="fa fa-check"></i><b>7.5</b> Clustering</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findsugestedk"><i class="fa fa-check"></i><b>7.5.1</b> findSugestedK()</a></li>
<li class="chapter" data-level="7.5.2" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#findclusters"><i class="fa fa-check"></i><b>7.5.2</b> findClusters()</a></li>
<li class="chapter" data-level="7.5.3" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotumap"><i class="fa fa-check"></i><b>7.5.3</b> plotUmap()</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runpca"><i class="fa fa-check"></i><b>7.6</b> runPca()</a></li>
<li class="chapter" data-level="7.7" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotheatmap"><i class="fa fa-check"></i><b>7.7</b> plotHeatmap()</a></li>
<li class="chapter" data-level="7.8" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#runphylo-plotphylo"><i class="fa fa-check"></i><b>7.8</b> runPhylo() &amp; plotPhylo()</a></li>
<li class="chapter" data-level="7.9" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#calcconsensus"><i class="fa fa-check"></i><b>7.9</b> calcConsensus()</a></li>
<li class="chapter" data-level="7.10" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotconsensusline"><i class="fa fa-check"></i><b>7.10</b> plotConsensusLine()</a></li>
<li class="chapter" data-level="7.11" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotgenecopy"><i class="fa fa-check"></i><b>7.11</b> plotGeneCopy()</a></li>
<li class="chapter" data-level="7.12" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotfreq"><i class="fa fa-check"></i><b>7.12</b> plotFreq()</a></li>
<li class="chapter" data-level="7.13" data-path="analysis-and-visualization-module.html"><a href="analysis-and-visualization-module.html#plotalluvial"><i class="fa fa-check"></i><b>7.13</b> plotAlluvial()</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="merging-copykit-objects.html"><a href="merging-copykit-objects.html"><i class="fa fa-check"></i><b>8</b> Merging CopyKit Objects</a></li>
<li class="chapter" data-level="9" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>9</b> Session Info</a></li>
<li class="chapter" data-level="10" data-path="faq.html"><a href="faq.html"><i class="fa fa-check"></i><b>10</b> FAQ</a>
<ul>
<li class="chapter" data-level="10.1" data-path="faq.html"><a href="faq.html#error-unable-to-load-package-after-installation."><i class="fa fa-check"></i><b>10.1</b> Error unable to load package after installation.</a></li>
<li class="chapter" data-level="10.2" data-path="faq.html"><a href="faq.html#copykit-is-very-slow-despite-parallel-processing"><i class="fa fa-check"></i><b>10.2</b> CopyKit is very slow despite parallel processing</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="learn-more.html"><a href="learn-more.html"><i class="fa fa-check"></i><b>11</b> Learn more</a>
<ul>
<li class="chapter" data-level="11.1" data-path="learn-more.html"><a href="learn-more.html#copy-number-data-from-short-reads-sequencing"><i class="fa fa-check"></i><b>11.1</b> Copy Number data from short reads sequencing</a></li>
<li class="chapter" data-level="11.2" data-path="learn-more.html"><a href="learn-more.html#principle-lm"><i class="fa fa-check"></i><b>11.2</b> Principle</a></li>
<li class="chapter" data-level="11.3" data-path="learn-more.html"><a href="learn-more.html#varbin-lm"><i class="fa fa-check"></i><b>11.3</b> VarBin</a></li>
<li class="chapter" data-level="11.4" data-path="learn-more.html"><a href="learn-more.html#gccor-lm"><i class="fa fa-check"></i><b>11.4</b> GC correction</a></li>
<li class="chapter" data-level="11.5" data-path="learn-more.html"><a href="learn-more.html#mergelevels-lm"><i class="fa fa-check"></i><b>11.5</b> Merge Levels</a></li>
<li class="chapter" data-level="11.6" data-path="learn-more.html"><a href="learn-more.html#infercn-lm"><i class="fa fa-check"></i><b>11.6</b> How do we infer copy numbers</a></li>
<li class="chapter" data-level="11.7" data-path="learn-more.html"><a href="learn-more.html#segmentation-lm"><i class="fa fa-check"></i><b>11.7</b> Segmentation</a>
<ul>
<li class="chapter" data-level="11.7.1" data-path="learn-more.html"><a href="learn-more.html#cbs-lm"><i class="fa fa-check"></i><b>11.7.1</b> CBS</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">CopyKit User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Session Info<a href="session-info.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="session-info.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.0 (2022-04-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils    
## [6] datasets  methods   base     
## 
## other attached packages:
##  [1] BiocParallel_1.30.2        
##  [2] copykit_0.1.2              
##  [3] DNAcopy_1.70.0             
##  [4] Rsubread_2.10.0            
##  [5] SingleCellExperiment_1.18.0
##  [6] SummarizedExperiment_1.26.1
##  [7] Biobase_2.56.0             
##  [8] GenomicRanges_1.48.0       
##  [9] GenomeInfoDb_1.32.2        
## [10] IRanges_2.30.0             
## [11] S4Vectors_0.34.0           
## [12] BiocGenerics_0.42.0        
## [13] MatrixGenerics_1.8.0       
## [14] matrixStats_0.62.0         
## 
## loaded via a namespace (and not attached):
##   [1] rappdirs_0.3.3            rtracklayer_1.56.0       
##   [3] prabclus_2.3-2            coda_0.19-4              
##   [5] tidyr_1.2.0               ggplot2_3.3.6            
##   [7] clusterGeneration_1.3.7   bit64_4.0.5              
##   [9] knitr_1.39                irlba_2.3.5              
##  [11] DelayedArray_0.22.0       KEGGREST_1.36.0          
##  [13] RCurl_1.98-1.6            doParallel_1.0.17        
##  [15] generics_0.1.2            GenomicFeatures_1.48.1   
##  [17] ScaledMatrix_1.4.0        cowplot_1.1.1            
##  [19] RSQLite_2.2.14            combinat_0.0-8           
##  [21] bit_4.0.4                 xml2_1.3.3               
##  [23] lubridate_1.8.0           httpuv_1.6.5             
##  [25] assertthat_0.2.1          viridis_0.6.2            
##  [27] amap_0.8-18               xfun_0.31                
##  [29] jquerylib_0.1.4           hms_1.1.1                
##  [31] ggdist_3.1.1              evaluate_0.15            
##  [33] promises_1.2.0.1          DEoptimR_1.0-11          
##  [35] fansi_1.0.3               restfulr_0.0.13          
##  [37] progress_1.2.2            dbplyr_2.1.1             
##  [39] igraph_1.3.1              DBI_1.1.2                
##  [41] tmvnsim_1.0-2             paletteer_1.4.0          
##  [43] purrr_0.3.4               ellipsis_0.3.2           
##  [45] dplyr_1.0.9               ggnewscale_0.4.7         
##  [47] backports_1.4.1           bookdown_0.26            
##  [49] biomaRt_2.52.0            sparseMatrixStats_1.8.0  
##  [51] vctrs_0.4.1               ggalluvial_0.12.3        
##  [53] Cairo_1.5-15              abind_1.4-5              
##  [55] withr_2.5.0               cachem_1.0.6             
##  [57] robustbase_0.95-0         GenomicAlignments_1.32.0 
##  [59] treeio_1.20.0             prettyunits_1.1.1        
##  [61] scran_1.24.0              mclust_5.4.9             
##  [63] mnormt_2.0.2              phytools_1.0-3           
##  [65] cluster_2.1.3             segmented_1.5-0          
##  [67] ape_5.6-2                 lazyeval_0.2.2           
##  [69] crayon_1.5.1              labeling_0.4.2           
##  [71] edgeR_3.38.1              pkgconfig_2.0.3          
##  [73] vipor_0.4.5               nlme_3.1-157             
##  [75] nnet_7.3-17               pals_1.7                 
##  [77] rlang_1.0.2               diptest_0.76-0           
##  [79] miniUI_0.1.1.1            lifecycle_1.0.1          
##  [81] filelock_1.0.2            dbscan_1.1-10            
##  [83] BiocFileCache_2.4.0       rsvd_1.0.5               
##  [85] dichromat_2.0-0.1         distributional_0.3.0     
##  [87] Matrix_1.5-3              aplot_0.1.4              
##  [89] phangorn_2.8.1            carData_3.0-5            
##  [91] boot_1.3-28               beeswarm_0.4.0           
##  [93] GlobalOptions_0.1.2       viridisLite_0.4.0        
##  [95] png_0.1-7                 rjson_0.2.21             
##  [97] bitops_1.0-7              Biostrings_2.64.0        
##  [99] blob_1.2.3                DelayedMatrixStats_1.18.0
## [101] shape_1.4.6               stringr_1.4.0            
## [103] rstatix_0.7.0             gridGraphics_0.5-1       
## [105] beachmat_2.12.0           scales_1.2.0             
## [107] memoise_2.0.1             magrittr_2.0.3           
## [109] zlibbioc_1.42.0           compiler_4.2.0           
## [111] dqrng_0.3.0               BiocIO_1.6.0             
## [113] RColorBrewer_1.1-3        plotrix_3.8-2            
## [115] clue_0.3-60               Rsamtools_2.12.0         
## [117] snakecase_0.11.0          cli_3.3.0                
## [119] XVector_0.36.0            patchwork_1.1.1          
## [121] MASS_7.3-57               tidyselect_1.1.2         
## [123] stringi_1.7.6             forcats_0.5.1            
## [125] highr_0.9                 purler_0.1.1             
## [127] yaml_2.3.5                BiocSingular_1.12.0      
## [129] locfit_1.5-9.5            grid_4.2.0               
## [131] sass_0.4.1                fastmatch_1.1-3          
## [133] tools_4.2.0               parallel_4.2.0           
## [135] circlize_0.4.15           rstudioapi_0.13          
## [137] bluster_1.6.0             foreach_1.5.2            
## [139] gridExtra_2.3             janitor_2.1.0            
## [141] metapod_1.4.0             scatterplot3d_0.3-41     
## [143] farver_2.1.0              digest_0.6.29            
## [145] FNN_1.1.3                 shiny_1.7.1              
## [147] quadprog_1.5-8            fpc_2.2-9                
## [149] Rcpp_1.0.8.3              car_3.0-13               
## [151] broom_0.8.0               scuttle_1.6.2            
## [153] later_1.3.0               httr_1.4.3               
## [155] AnnotationDbi_1.58.0      ComplexHeatmap_2.12.0    
## [157] kernlab_0.9-30            colorspace_2.0-3         
## [159] XML_3.99-0.9              reticulate_1.25          
## [161] splines_4.2.0             uwot_0.1.11              
## [163] yulab.utils_0.0.4         statmod_1.4.36           
## [165] rematch2_2.1.2            tidytree_0.3.9           
## [167] expm_0.999-6              mapproj_1.2.8            
## [169] ggplotify_0.1.0           flexmix_2.3-17           
## [171] xtable_1.8-4              jsonlite_1.8.0           
## [173] ggtree_3.4.0              modeltools_0.2-23        
## [175] ggfun_0.0.6               R6_2.5.1                 
## [177] pillar_1.7.0              htmltools_0.5.2          
## [179] mime_0.12                 glue_1.6.2               
## [181] fastmap_1.1.0             BiocNeighbors_1.14.0     
## [183] class_7.3-20              codetools_0.2-18         
## [185] maps_3.4.0                utf8_1.2.2               
## [187] bslib_0.3.1               lattice_0.20-45          
## [189] tibble_3.1.7              mixtools_1.2.0           
## [191] numDeriv_2016.8-1.1       ggbeeswarm_0.6.0         
## [193] scquantum_1.0.0           curl_4.3.2               
## [195] leiden_0.4.2              gtools_3.9.2             
## [197] magick_2.7.3              survival_3.3-1           
## [199] limma_3.52.1              rmarkdown_2.14           
## [201] munsell_0.5.0             GetoptLong_1.0.5         
## [203] fastcluster_1.2.3         GenomeInfoDbData_1.2.8   
## [205] iterators_1.0.14          gtable_0.3.0</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="merging-copykit-objects.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="faq.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CopyKit-UserGuide.pdf", "CopyKit-UserGuide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
